{
  "nbformat": 4,
  "nbformat_minor": 0,
  "metadata": {
    "colab": {
      "provenance": []
    },
    "kernelspec": {
      "name": "python3",
      "display_name": "Python 3"
    },
    "language_info": {
      "name": "python"
    }
  },
  "cells": [
    {
      "cell_type": "code",
      "source": [
        "!pip install azure-identity\n",
        "!pip install azure-search-documents"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "EqcrjXwDc4Nr",
        "outputId": "e8fdb180-7c69-4564-c492-014f34fc4bbd"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Collecting azure-identity\n",
            "  Downloading azure_identity-1.14.0-py3-none-any.whl (160 kB)\n",
            "\u001b[?25l     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m0.0/160.2 kB\u001b[0m \u001b[31m?\u001b[0m eta \u001b[36m-:--:--\u001b[0m\r\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m160.2/160.2 kB\u001b[0m \u001b[31m5.6 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hCollecting azure-core<2.0.0,>=1.11.0 (from azure-identity)\n",
            "  Downloading azure_core-1.29.1-py3-none-any.whl (188 kB)\n",
            "\u001b[?25l     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m0.0/188.5 kB\u001b[0m \u001b[31m?\u001b[0m eta \u001b[36m-:--:--\u001b[0m\r\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m188.5/188.5 kB\u001b[0m \u001b[31m20.2 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hRequirement already satisfied: cryptography>=2.5 in /usr/local/lib/python3.10/dist-packages (from azure-identity) (41.0.3)\n",
            "Collecting msal<2.0.0,>=1.20.0 (from azure-identity)\n",
            "  Downloading msal-1.23.0-py2.py3-none-any.whl (90 kB)\n",
            "\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m90.8/90.8 kB\u001b[0m \u001b[31m9.8 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hCollecting msal-extensions<2.0.0,>=0.3.0 (from azure-identity)\n",
            "  Downloading msal_extensions-1.0.0-py2.py3-none-any.whl (19 kB)\n",
            "Requirement already satisfied: requests>=2.18.4 in /usr/local/lib/python3.10/dist-packages (from azure-core<2.0.0,>=1.11.0->azure-identity) (2.31.0)\n",
            "Requirement already satisfied: six>=1.11.0 in /usr/local/lib/python3.10/dist-packages (from azure-core<2.0.0,>=1.11.0->azure-identity) (1.16.0)\n",
            "Requirement already satisfied: typing-extensions>=4.3.0 in /usr/local/lib/python3.10/dist-packages (from azure-core<2.0.0,>=1.11.0->azure-identity) (4.7.1)\n",
            "Requirement already satisfied: cffi>=1.12 in /usr/local/lib/python3.10/dist-packages (from cryptography>=2.5->azure-identity) (1.15.1)\n",
            "Requirement already satisfied: PyJWT[crypto]<3,>=1.0.0 in /usr/lib/python3/dist-packages (from msal<2.0.0,>=1.20.0->azure-identity) (2.3.0)\n",
            "Collecting portalocker<3,>=1.0 (from msal-extensions<2.0.0,>=0.3.0->azure-identity)\n",
            "  Downloading portalocker-2.7.0-py2.py3-none-any.whl (15 kB)\n",
            "Requirement already satisfied: pycparser in /usr/local/lib/python3.10/dist-packages (from cffi>=1.12->cryptography>=2.5->azure-identity) (2.21)\n",
            "Requirement already satisfied: charset-normalizer<4,>=2 in /usr/local/lib/python3.10/dist-packages (from requests>=2.18.4->azure-core<2.0.0,>=1.11.0->azure-identity) (3.2.0)\n",
            "Requirement already satisfied: idna<4,>=2.5 in /usr/local/lib/python3.10/dist-packages (from requests>=2.18.4->azure-core<2.0.0,>=1.11.0->azure-identity) (3.4)\n",
            "Requirement already satisfied: urllib3<3,>=1.21.1 in /usr/local/lib/python3.10/dist-packages (from requests>=2.18.4->azure-core<2.0.0,>=1.11.0->azure-identity) (1.26.16)\n",
            "Requirement already satisfied: certifi>=2017.4.17 in /usr/local/lib/python3.10/dist-packages (from requests>=2.18.4->azure-core<2.0.0,>=1.11.0->azure-identity) (2023.7.22)\n",
            "Installing collected packages: portalocker, azure-core, msal, msal-extensions, azure-identity\n",
            "Successfully installed azure-core-1.29.1 azure-identity-1.14.0 msal-1.23.0 msal-extensions-1.0.0 portalocker-2.7.0\n",
            "Collecting azure-search-documents\n",
            "  Downloading azure_search_documents-11.3.0-py3-none-any.whl (244 kB)\n",
            "\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m244.1/244.1 kB\u001b[0m \u001b[31m6.5 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hRequirement already satisfied: azure-core<2.0.0,>=1.19.0 in /usr/local/lib/python3.10/dist-packages (from azure-search-documents) (1.29.1)\n",
            "Collecting msrest>=0.6.21 (from azure-search-documents)\n",
            "  Downloading msrest-0.7.1-py3-none-any.whl (85 kB)\n",
            "\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m85.4/85.4 kB\u001b[0m \u001b[31m9.5 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hCollecting azure-common~=1.1 (from azure-search-documents)\n",
            "  Downloading azure_common-1.1.28-py2.py3-none-any.whl (14 kB)\n",
            "Requirement already satisfied: typing-extensions>=3.7.4.3 in /usr/local/lib/python3.10/dist-packages (from azure-search-documents) (4.7.1)\n",
            "Requirement already satisfied: requests>=2.18.4 in /usr/local/lib/python3.10/dist-packages (from azure-core<2.0.0,>=1.19.0->azure-search-documents) (2.31.0)\n",
            "Requirement already satisfied: six>=1.11.0 in /usr/local/lib/python3.10/dist-packages (from azure-core<2.0.0,>=1.19.0->azure-search-documents) (1.16.0)\n",
            "Requirement already satisfied: certifi>=2017.4.17 in /usr/local/lib/python3.10/dist-packages (from msrest>=0.6.21->azure-search-documents) (2023.7.22)\n",
            "Collecting isodate>=0.6.0 (from msrest>=0.6.21->azure-search-documents)\n",
            "  Downloading isodate-0.6.1-py2.py3-none-any.whl (41 kB)\n",
            "\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m41.7/41.7 kB\u001b[0m \u001b[31m5.0 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hRequirement already satisfied: requests-oauthlib>=0.5.0 in /usr/local/lib/python3.10/dist-packages (from msrest>=0.6.21->azure-search-documents) (1.3.1)\n",
            "Requirement already satisfied: charset-normalizer<4,>=2 in /usr/local/lib/python3.10/dist-packages (from requests>=2.18.4->azure-core<2.0.0,>=1.19.0->azure-search-documents) (3.2.0)\n",
            "Requirement already satisfied: idna<4,>=2.5 in /usr/local/lib/python3.10/dist-packages (from requests>=2.18.4->azure-core<2.0.0,>=1.19.0->azure-search-documents) (3.4)\n",
            "Requirement already satisfied: urllib3<3,>=1.21.1 in /usr/local/lib/python3.10/dist-packages (from requests>=2.18.4->azure-core<2.0.0,>=1.19.0->azure-search-documents) (1.26.16)\n",
            "Requirement already satisfied: oauthlib>=3.0.0 in /usr/local/lib/python3.10/dist-packages (from requests-oauthlib>=0.5.0->msrest>=0.6.21->azure-search-documents) (3.2.2)\n",
            "Installing collected packages: azure-common, isodate, msrest, azure-search-documents\n",
            "Successfully installed azure-common-1.1.28 azure-search-documents-11.3.0 isodate-0.6.1 msrest-0.7.1\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "-X8viP-ScYzh"
      },
      "outputs": [],
      "source": [
        "import os\n",
        "from azure.core.credentials import AzureKeyCredential\n",
        "from azure.search.documents import SearchClient\n",
        "\n",
        "#pip install azure-search-documents\n",
        "#pip install azure-core\n",
        "index_name = \"** Enter your index name\"\n",
        "# Get the service endpoint and API key from the environment\n",
        "endpoint = \"** Enter your endpoint\"\n",
        "key = \"** Enter your key \"\n",
        "\n",
        "# Create a client\n",
        "credential = AzureKeyCredential(key)\n",
        "client = SearchClient(endpoint=endpoint,\n",
        "                      index_name=index_name,\n",
        "                      credential=credential)"
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "results = client.search(search_text=\"How does liver cancer originate?\")\n",
        "finalresult = []\n",
        "\n",
        "# turn this into a function\n",
        "for result in results:\n",
        "  # creating a temporary dictionary where we are adding each filepatha and searchscore to the list\n",
        "  mydict = {}\n",
        "  mydict[\"filepath\"] = result[\"filepath\"]\n",
        "  mydict[\"@search.score\"] = result[\"@search.score\"]\n",
        "\n",
        "  # print(result[\"content\"])\n",
        "\n",
        "  finalresult.append(mydict)\n",
        "print(finalresult)\n",
        "\n",
        "  # [ {\"filePath\":11,\"score\":11}, {},{}]\n",
        "\n"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "Bqn4EOLLce7g",
        "outputId": "d5d8f388-f4ca-483f-a964-f02c0387bcac"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "[{'filepath': 'pubmed-33080466.txt', '@search.score': 3.1448805}, {'filepath': 'pubmed-35263795.txt', '@search.score': 2.9881814}, {'filepath': 'pubmed-31378681.txt', '@search.score': 2.3761964}, {'filepath': 'pubmed-36158262.txt', '@search.score': 1.8384137}, {'filepath': 'pubmed-31378681.txt', '@search.score': 1.4379532}, {'filepath': 'pubmed-31378681.txt', '@search.score': 1.2626619}, {'filepath': 'pubmed-32111216.txt', '@search.score': 1.1230905}, {'filepath': 'pubmed-32045030.txt', '@search.score': 1.0495409}, {'filepath': 'Cognitive Science Final Paper.docx', '@search.score': 0.9617383}, {'filepath': 'pubmed-7578390.txt', '@search.score': 0.9507718}, {'filepath': 'Written Report First Draft.docx', '@search.score': 0.8859299}, {'filepath': 'pubmed-32402299.txt', '@search.score': 0.8788067}, {'filepath': 'Saumya_Soni_Latest_Resume.pdf', '@search.score': 0.8145475}, {'filepath': 'Written Report First Draft.docx', '@search.score': 0.8102636}, {'filepath': 'Written Report First Draft.docx', '@search.score': 0.72675514}, {'filepath': 'pubmed-31378681.txt', '@search.score': 0.7245507}, {'filepath': 'Cognitive Science Final Paper.docx', '@search.score': 0.63890904}, {'filepath': 'pubmed-26187874.txt', '@search.score': 0.63370943}, {'filepath': 'pubmed-31666728.txt', '@search.score': 0.5750315}, {'filepath': 'pubmed-36158262.txt', '@search.score': 0.5269685}, {'filepath': 'Cognitive Science Final Paper.docx', '@search.score': 0.49963242}, {'filepath': 'summary-31378681.txt', '@search.score': 0.4767133}, {'filepath': 'pubmed-31666728.txt', '@search.score': 0.40088063}]\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "def userquery(question):\n",
        "  results = client.search(search_text=question)\n",
        "  finalresult = []\n",
        "\n",
        "  # turn this into a function\n",
        "  for result in results:\n",
        "  # creating a temporary dictionary where we are adding each filepath and searchscore to the list\n",
        "    mydict = {}\n",
        "    mydict[\"filepath\"] = result[\"filepath\"]\n",
        "    mydict[\"content\"] = result[\"content\"]\n",
        "    mydict[\"@search.score\"] = result[\"@search.score\"]\n",
        "    finalresult.append(mydict)\n",
        "\n",
        "\n",
        "  # print(result[\"content\"])\n",
        "\n",
        "  return finalresult\n"
      ],
      "metadata": {
        "id": "6KL6Lew_dAuc"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "userquery(\"How does liver cancer originate\")"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "KDpRvxxceeb_",
        "outputId": "a8e4ba03-da7b-4786-f710-718edb1aa8ab"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "[{'filepath': 'pubmed-33080466.txt',\n",
              "  'content': 'PMID- 33080466\\r\\nOWN - NLM\\r\\nSTAT- MEDLINE\\r\\nDCOM- 20210219\\r\\nLR - 20210219\\r\\nIS - 1950-6007 (Electronic)\\r\\nIS - 0753-3322 (Linking)\\r\\nVI - 132\\r\\nDP - 2020 Dec\\r\\nTI - WNT/β-catenin signaling in the development of liver cancers.\\r\\nPG - 110851\\r\\nLID - S0753-3322(20)31043-X [pii]\\r\\nLID - 10.1016/j.biopha.2020.110851 [doi]\\r\\nAB - The WNT/β-catenin signaling pathway is a highly conserved and tightly controlled \\n molecular mechanism that regulates embryonic development, cellular proliferation \\n and differentiation. Of note, accumulating evidence has shown that the aberrant \\n of WNT/β-catenin signaling promotes the development and/or progression of liver \\n cancer, including hepatocellular carcinoma (HCC) and cholangiocarcinoma (CCA), \\n the two most prevalent primary liver tumours in adults. There are two different \\n WNT signaling pathways have been identified, which were termed non-canonical and \\n canonical pathways, the latter involving the activation of β-catenin. β-catenin, \\n acting as an intracellular signal transducer in the WNT signaling pathway, is \\n encoded by CTNNB1 and plays a critical role in tumorigenesis. In the past \\n research, most liver tumors have mutations in genes encoding key components of \\n the WNT/β-catenin signaling pathway. In addition, several of other signaling \\n pathways also can crosswalk with β-catenin. In this review, we discuss the most \\n relevant molecular mechanisms of action and regulation of WNT/β-catenin signaling \\n in the development and pathophysiology of liver cancers, as well as in the \\n development of therapeutics.\\r\\nCI - Copyright © 2020 The Author(s). Published by Elsevier Masson SAS.. All rights \\n reserved.\\r\\nFAU - He, Shuai\\r\\nAU - He S\\r\\nAD - Department of General Surgery, the Fourth Affiliated Hospital, China Medical \\n University, Shenyang 110032, China.\\r\\nFAU - Tang, Shilei\\r\\nAU - Tang S\\r\\nAD - Department of General Surgery, the Fourth Affiliated Hospital, China Medical \\n University, Shenyang 110032, China. Electronic address: sltang@cmu.edu.cn.\\r\\nLA - eng\\r\\nPT - Journal Article\\r\\nPT - Review\\r\\nDEP - 20201017\\r\\nPL - France\\r\\nTA - Biomed Pharmacother\\r\\nJT - Biomedicine & pharmacotherapy = Biomedecine & pharmacotherapie\\r\\nJID - 8213295\\r\\nRN - 0 (CTNNB1 protein, human)\\r\\nRN - 0 (beta Catenin)\\r\\nSB - IM\\r\\nMH - Adult\\r\\nMH - Animals\\r\\nMH - Bile Duct Neoplasms/genetics/pathology\\r\\nMH - Carcinogenesis/genetics\\r\\nMH - Carcinoma, Hepatocellular/genetics/pathology\\r\\nMH - Cholangiocarcinoma/genetics\\r\\nMH - Humans\\r\\nMH - Liver Neoplasms/genetics/*pathology\\r\\nMH - Mutation\\r\\nMH - Wnt Signaling Pathway/*genetics\\r\\nMH - beta Catenin/*genetics\\r\\nOTO - NOTNLM\\r\\nOT - Liver cancer\\r\\nOT - Therapy\\r\\nOT - WNT/β-catenin\\r\\nEDAT- 2020/10/21 06:00\\r\\nMHDA- 2021/02/20 06:00\\r\\nCRDT- 2020/10/20 20:11\\r\\nPHST- 2020/08/26 00:00 [received]\\r\\nPHST- 2020/09/27 00:00 [revised]\\r\\nPHST- 2020/10/04 00:00 [accepted]\\r\\nPHST- 2020/10/21 06:00 [pubmed]\\r\\nPHST- 2021/02/20 06:00 [medline]\\r\\nPHST- 2020/10/20 20:11 [entrez]\\r\\nAID - S0753-3322(20)31043-X [pii]\\r\\nAID - 10.1016/j.biopha.2020.110851 [doi]\\r\\nPST - ppublish\\r\\nSO - Biomed Pharmacother',\n",
              "  '@search.score': 3.1448805},\n",
              " {'filepath': 'pubmed-35263795.txt',\n",
              "  'content': \"PMID- 35263795\\r\\nOWN - NLM\\r\\nSTAT- MEDLINE\\r\\nDCOM- 20220627\\r\\nLR - 20220630\\r\\nIS - 1098-8971 (Electronic)\\r\\nIS - 0272-8087 (Linking)\\r\\nVI - 42\\r\\nIP - 2\\r\\nDP - 2022 May\\r\\nTI - Liver Immunology, Immunotherapy, and Liver Cancers: Time for a Rethink?\\r\\nPG - 212-224\\r\\nLID - 10.1055/s-0042-1744143 [doi]\\r\\nAB - The complex immune system of the liver has a major role in tumor surveillance, \\n but also partly explains why current immune therapies are poorly effective \\n against liver cancers. Known primarily for its tolerogenic capacity, the hepatic \\n immune repertoire also comprises diverse populations of armored immune cells with \\n tumor surveillant roles. In healthy people, these work together to successfully \\n identify malignant cells and prevent their proliferation, thus halting tumor \\n formation. When frontline hepatic immune surveillance systems fail, compromised \\n hepatic immunity, driven by obesity, infection, or other pathological factors, \\n allows primary or secondary liver cancers to develop. Tumor growth promotes the \\n normal tolerogenic immunological milieu of the liver, perhaps explaining why \\n current immunotherapies fail to work. This review explores the complex local \\n liver immune system with the hope of identifying potential therapeutic targets \\n needed to best overcome immunological barriers in the liver to create an \\n environment no longer hostile to immunotherapy for the treatment of liver cancer.\\r\\nCI - Thieme. All rights reserved.\\r\\nFAU - Carroll, Hailey K\\r\\nAU - Carroll HK\\r\\nAD - Department of Medical Oncology, The Mater Hospital, Dublin, Ireland.\\r\\nFAU - Duffy, Austin G\\r\\nAU - Duffy AG\\r\\nAD - Department of Medical Oncology, The Mater Hospital, Dublin, Ireland.\\r\\nFAU - O'Farrelly, Cliona\\r\\nAU - O'Farrelly C\\r\\nAD - School of Biochemistry and Immunology, Trinity Biomedical Sciences Institute, \\n Trinity College, Dublin, Ireland.\\r\\nAD - School of Medicine, Trinity Biomedical Sciences Institute, Trinity College, \\n Dublin, Ireland.\\r\\nLA - eng\\r\\nPT - Journal Article\\r\\nPT - Review\\r\\nDEP - 20220309\\r\\nPL - United States\\r\\nTA - Semin Liver Dis\\r\\nJT - Seminars in liver disease\\r\\nJID - 8110297\\r\\nSB - IM\\r\\nMH - Humans\\r\\nMH - *Immunotherapy\\r\\nMH - *Liver Neoplasms/therapy\\r\\nMH - Tumor Microenvironment\\r\\nCOIS- None declared.\\r\\nEDAT- 2022/03/10 06:00\\r\\nMHDA- 2022/06/28 06:00\\r\\nCRDT- 2022/03/09 20:08\\r\\nPHST- 2022/03/10 06:00 [pubmed]\\r\\nPHST- 2022/06/28 06:00 [medline]\\r\\nPHST- 2022/03/09 20:08 [entrez]\\r\\nAID - 10.1055/s-0042-1744143 [doi]\\r\\nPST - ppublish\\r\\nSO - Semin Liver Dis. 2022 May;42(2):212-224. doi: 10.1055/s-0042-1744143. Epub 2022 \\n Mar 9.\",\n",
              "  '@search.score': 2.9881814},\n",
              " {'filepath': 'pubmed-31378681.txt',\n",
              "  'content': \"FAU - Zhang, Zhigang\\r\\nAU - Zhang Z\\r\\nAD - Department of Epidemiology and Biostatistics, Memorial Sloan-Kettering Cancer \\n Center, New York, NY 10065, USA.\\r\\nFAU - Meng, Zhiqiang\\r\\nAU - Meng Z\\r\\nAD - Department of Minimally Invasive Therapy, Collaborative Innovation Center for \\n Cancer Medicine, Fudan University Shanghai Cancer Center, Department of Oncology, \\n Shanghai Medical College, Fudan University, Shanghai 200032, China.\\r\\nFAU - Shi, Xiaolei\\r\\nAU - Shi X\\r\\nAD - Department of Hepatobiliary Surgery, the Affiliated Drum Tower Hospital, Medical \\n School of Nanjing University, Nanjing, Jiangsu 211166, China. Electronic address: \\n njsxl2000@163.com.\\r\\nFAU - Zhang, Haibin\\r\\nAU - Zhang H\\r\\nAD - Fifth Department of Hepatic Surgery, Eastern Hepatobiliary Surgery Hospital, \\n Second Military Medical University, Shanghai 200438, China. Electronic address: \\n drzhanghb@163.com.\\r\\nFAU - Li, Yixue\\r\\nAU - Li Y\\r\\nAD - CAS Key Laboratory of Computational Biology, CAS-MPG Partner Institute for \\n Computational Biology, Shanghai Institute of Nutrition and Health, Shanghai \\n Institutes for Biological Sciences, University of Chinese Academy of Sciences, \\n Chinese Academy of Sciences, Shanghai 200031, China. Electronic address: \\n yxli@sibs.ac.cn.\\r\\nFAU - Hui, Lijian\\r\\nAU - Hui L\\r\\nAD - State Key Laboratory of Cell Biology, CAS Center for Excellence in Molecular Cell \\n Science, Shanghai Institute of Biochemistry and Cell Biology, University of \\n Chinese Academy of Sciences, Chinese Academy of Sciences, Shanghai 200031, China; \\n School of Life Science and Technology, Shanghai Tech University, Shanghai 201210, \\n China; Bio-Research Innovation Center Suzhou, Shanghai Institute of Biochemistry \\n and Cell Biology, Chinese Academy of Sciences, Suzhou, Jiangsu 215121, China. \\n Electronic address: ljhui@sibcb.ac.cn.\\r\\nLA - eng\\r\\nGR - P30 CA008748/CA/NCI NIH HHS/United States\\r\\nPT - Journal Article\\r\\nPT - Research Support, Non-U.S. Gov't\\r\\nDEP - 20190801\\r\\nPL - United States\\r\\nTA - Cancer Cell\\r\\nJT - Cancer cell\\r\\nJID - 101130617\\r\\nRN - 0 (Antineoplastic Agents)\\r\\nRN - 0 (Biomarkers, Tumor)\\r\\nRN - 0 (Protein Kinase Inhibitors)\\r\\nRN - 9ZOQ3TZI87 (Sorafenib)\\r\\nSB - IM\\r\\nMH - Animals\\r\\nMH - Antineoplastic Agents/*pharmacology\\r\\nMH - Asian People/genetics\\r\\nMH - Biomarkers, Tumor/*genetics\\r\\nMH - Carcinoma, Hepatocellular/*drug therapy/ethnology/*genetics/pathology\\r\\nMH - Cell Line, Tumor\\r\\nMH - Clinical Decision-Making\\r\\nMH - Databases, Genetic\\r\\nMH - Drug Resistance, Neoplasm/genetics\\r\\nMH - Female\\r\\nMH - Genetic Heterogeneity\\r\\nMH - Genetic Predisposition to Disease\\r\\nMH - High-Throughput Nucleotide Sequencing\\r\\nMH - Humans\\r\\nMH - Liver Neoplasms/*drug therapy/ethnology/*genetics/pathology\\r\\nMH - Male\\r\\nMH - Mice, Inbred BALB C\\r\\nMH - Mice, Inbred NOD\\r\\nMH - Mice, Nude\\r\\nMH - Mice, SCID\\r\\nMH - Patient Selection\\r\\nMH - Pharmacogenomic Testing\\r\\nMH - *Pharmacogenomic Variants\\r\\nMH - Phenotype\\r\\nMH - Precision Medicine\\r\\nMH - Protein Kinase Inhibitors/*pharmacology\\r\\nMH - Sorafenib/*pharmacology\\r\\nMH - Xenograft Model Antitumor Assays\\r\\nPMC - PMC7505724\\r\\nMID - NIHMS1622926\\r\\nOTO - NOTNLM\\r\\nOT - liver cancer\\r\\nOT - patient-derived cancer models\\r\\nOT - pharmacogenomics\\r\\nOT - sorafenib\\r\\nCOIS- DECLARATION OF INTERESTS He Zhou, Wei Wang, Fubo Xie and Ying Gu are employees of \\n Shanghai ChemPartner Co., Ltd.\",\n",
              "  '@search.score': 2.3761964},\n",
              " {'filepath': 'pubmed-36158262.txt',\n",
              "  'content': 'LA - eng\\r\\nPT - Journal Article\\r\\nPT - Review\\r\\nPL - United States\\r\\nTA - World J Gastroenterol\\r\\nJT - World journal of gastroenterology\\r\\nJID - 100883448\\r\\nSB - IM\\r\\nMH - Artificial Intelligence\\r\\nMH - *Carcinoma, Hepatocellular/diagnostic imaging/therapy\\r\\nMH - Humans\\r\\nMH - *Liver Neoplasms/diagnostic imaging/therapy\\r\\nMH - Ultrasonography/methods\\r\\nPMC - PMC9346461\\r\\nOTO - NOTNLM\\r\\nOT - Deep learning\\r\\nOT - Diffuse liver diseases\\r\\nOT - Focal liver diseases\\r\\nOT - Machine learning\\r\\nOT - Radiomics\\r\\nOT - Ultrasound\\r\\nCOIS- Conflict-of-interest statement: All the authors report no relevant conflicts of \\n interest for this article.\\r\\nEDAT- 2022/09/27 06:00\\r\\nMHDA- 2022/09/28 06:00\\r\\nCRDT- 2022/09/26 17:18\\r\\nPHST- 2022/02/07 00:00 [received]\\r\\nPHST- 2022/04/18 00:00 [revised]\\r\\nPHST- 2022/06/19 00:00 [accepted]\\r\\nPHST- 2022/09/26 17:18 [entrez]\\r\\nPHST- 2022/09/27 06:00 [pubmed]\\r\\nPHST- 2022/09/28 06:00 [medline]\\r\\nAID - 10.3748/wjg.v28.i27.3398 [doi]\\r\\nPST - ppublish\\r\\nSO - World J Gastroenterol. 2022 Jul 21;28(27):3398-3409. doi: \\n 10.3748/wjg.v28.i27.3398.',\n",
              "  '@search.score': 1.8384137},\n",
              " {'filepath': 'pubmed-31378681.txt',\n",
              "  'content': '. The remaining authors declare no competing \\n financial interests.\\r\\nEDAT- 2019/08/06 06:00\\r\\nMHDA- 2020/04/21 06:00\\r\\nCRDT- 2019/08/06 06:00\\r\\nPHST- 2018/11/27 00:00 [received]\\r\\nPHST- 2019/05/17 00:00 [revised]\\r\\nPHST- 2019/07/01 00:00 [accepted]\\r\\nPHST- 2019/08/06 06:00 [pubmed]\\r\\nPHST- 2020/04/21 06:00 [medline]\\r\\nPHST- 2019/08/06 06:00 [entrez]\\r\\nAID - S1535-6108(19)30328-9 [pii]\\r\\nAID - 10.1016/j.ccell.2019.07.001 [doi]\\r\\nPST - ppublish\\r\\nSO - Cancer Cell. 2019 Aug 12;36(2):179-193.e11. doi: 10.1016/j.ccell.2019.07.001. \\n Epub 2019 Aug 1.',\n",
              "  '@search.score': 1.4379532},\n",
              " {'filepath': 'pubmed-31378681.txt',\n",
              "  'content': 'PMID- 31378681\\r\\nOWN - NLM\\r\\nSTAT- MEDLINE\\r\\nDCOM- 20200420\\r\\nLR - 20221207\\r\\nIS - 1878-3686 (Electronic)\\r\\nIS - 1535-6108 (Print)\\r\\nIS - 1535-6108 (Linking)\\r\\nVI - 36\\r\\nIP - 2\\r\\nDP - 2019 Aug 12\\r\\nTI - A Pharmacogenomic Landscape in Human Liver Cancers.\\r\\nPG - 179-193.e11\\r\\nLID - S1535-6108(19)30328-9 [pii]\\r\\nLID - 10.1016/j.ccell.2019.07.001 [doi]\\r\\nAB - Liver cancers are highly heterogeneous with poor prognosis and drug response. A \\n better understanding between genetic alterations and drug responses would \\n facilitate precision treatment for liver cancers. To characterize the landscape \\n of pharmacogenomic interactions in liver cancers, we developed a protocol to \\n establish human liver cancer cell models at a success rate of around 50% and \\n generated the Liver Cancer Model Repository (LIMORE) with 81 cell models. LIMORE \\n represented genomic and transcriptomic heterogeneity of primary cancers. \\n Interrogation of the pharmacogenomic landscape of LIMORE discovered unexplored \\n gene-drug associations, including synthetic lethalities to prevalent alterations \\n in liver cancers. Moreover, predictive biomarker candidates were suggested for \\n the selection of sorafenib-responding patients. LIMORE provides a rich resource \\n facilitating drug discovery in liver cancers.\\r\\nCI - Copyright © 2019 Elsevier Inc. All rights reserved.\\r\\nFAU - Qiu, Zhixin\\r\\nAU - Qiu Z\\r\\nAD - State Key Laboratory of Cell Biology, CAS Center for Excellence in Molecular Cell \\n Science, Shanghai Institute of Biochemistry and Cell Biology, University of \\n Chinese Academy of Sciences, Chinese Academy of Sciences, Shanghai 200031, China.\\r\\nFAU - Li, Hong\\r\\nAU - Li H\\r\\nAD - CAS Key Laboratory of Computational Biology, CAS-MPG Partner Institute for \\n Computational Biology, Shanghai Institute of Nutrition and Health, Shanghai \\n Institutes for Biological Sciences, University of Chinese Academy of Sciences, \\n Chinese Academy of Sciences, Shanghai 200031, China.\\r\\nFAU - Zhang, Zhengtao\\r\\nAU - Zhang Z\\r\\nAD - State Key Laboratory of Cell Biology, CAS Center for Excellence in Molecular Cell \\n Science, Shanghai Institute of Biochemistry and Cell Biology, University of \\n Chinese Academy of Sciences, Chinese Academy of Sciences, Shanghai 200031, China.\\r\\nFAU - Zhu, Zhenfeng\\r\\nAU - Zhu Z\\r\\nAD - Department of Minimally Invasive Therapy, Collaborative Innovation Center for \\n Cancer Medicine, Fudan University Shanghai Cancer Center, Department of Oncology, \\n Shanghai Medical College, Fudan University, Shanghai 200032, China.\\r\\nFAU - He, Sheng\\r\\nAU - He S\\r\\nAD - CAS Key Laboratory of Computational Biology, CAS-MPG Partner Institute for \\n Computational Biology, Shanghai Institute of Nutrition and Health, Shanghai \\n Institutes for Biological Sciences, University of Chinese Academy of Sciences, \\n Chinese Academy of Sciences, Shanghai 200031, China; School of Life Science and \\n Technology, Shanghai Tech University, Shanghai 201210, China.\\r\\nFAU - Wang, Xujun\\r\\nAU - Wang X\\r\\nAD - SJTU-Yale Joint Center for Biostatistics, Department of Bioinformatics and \\n Biostatistics, Shanghai Jiaotong University, Shanghai 200240, China.\\r\\nFAU - Wang, Pengcheng\\r\\nAU - Wang P\\r\\nAD - Liver Cancer Institute, Zhongshan Hospital, Fudan University, Key Laboratory of \\n Carcinogenesis and Cancer Invasion, Fudan University, Shanghai 200032, China.\\r\\nFAU - Qin, Jianjie\\r\\nAU - Qin J\\r\\nAD - Liver Transplantation Center, Key Laboratory of Living Donor Liver \\n Transplantation of Ministry of Public Health, The First Affiliated Hospital of \\n Nanjing Medical University, Nanjing, Jiangsu 210029, China.\\r\\nFAU - Zhuang, Liping\\r\\nAU - Zhuang L\\r\\nAD - Department of Minimally Invasive Therapy, Collaborative Innovation Center for \\n Cancer Medicine, Fudan University Shanghai Cancer Center, Department of Oncology, \\n Shanghai Medical College, Fudan University, Shanghai 200032, China.\\r\\nFAU - Wang, Wei\\r\\nAU - Wang W\\r\\nAD - Shanghai ChemPartner Co., Ltd., Shanghai 201203, China',\n",
              "  '@search.score': 1.2626619},\n",
              " {'filepath': 'pubmed-32111216.txt',\n",
              "  'content': \"PMID- 32111216\\r\\nOWN - NLM\\r\\nSTAT- MEDLINE\\r\\nDCOM- 20210201\\r\\nLR - 20210806\\r\\nIS - 1476-4598 (Electronic)\\r\\nIS - 1476-4598 (Linking)\\r\\nVI - 19\\r\\nIP - 1\\r\\nDP - 2020 Feb 28\\r\\nTI - The emerging roles of N6-methyladenosine (m6A) deregulation in liver \\n carcinogenesis.\\r\\nPG - 44\\r\\nLID - 10.1186/s12943-020-01172-y [doi]\\r\\nLID - 44\\r\\nAB - Liver cancer is a common cancer worldwide. Although the etiological factors of \\n liver carcinogenesis are well defined, the underlying molecular mechanisms remain \\n largely elusive. Epigenetic deregulations, such as aberrant DNA methylation and \\n histone modifications, play a critical role in liver carcinogenesis. Analogous to \\n DNA and core histone proteins, reversible chemical modifications on mRNA have \\n recently been recognized as important regulatory mechanisms to control gene \\n expression. N6-methyladenosine (m6A) is the most prevalent internal mRNA \\n modification in mammalian cells. m6A modification is important for controlling \\n many cellular and biological processes. Deregulation of m6A modification has been \\n recently implicated in human carcinogenesis, including liver cancer. In this \\n review, we summarize the recent findings on m6A regulation and its biological \\n impacts in normal and cancer cells. We will focus on the deregulation of m6A \\n modification and m6A regulators in liver diseases and liver cancers. We will \\n highlight the clinical relevance of m6A deregulation in liver cancer. We will \\n also discuss the potential of exploiting m6A modification for cancer diagnosis \\n and therapeutics.\\r\\nFAU - Chen, Mengnuo\\r\\nAU - Chen M\\r\\nAD - State Key Laboratory of Liver Research, the University of Hong Kong, Hong Kong, \\n China.\\r\\nFAU - Wong, Chun-Ming\\r\\nAU - Wong CM\\r\\nAD - Department of Pathology, Li Ka Shing Faculty of Medicine, the University of Hong \\n Kong, Hong Kong, China. jackwong@pathology.hku.hk.\\r\\nLA - eng\\r\\nPT - Journal Article\\r\\nPT - Research Support, Non-U.S. Gov't\\r\\nPT - Review\\r\\nDEP - 20200228\\r\\nPL - England\\r\\nTA - Mol Cancer\\r\\nJT - Molecular cancer\\r\\nJID - 101147698\\r\\nRN - 0 (Biomarkers, Tumor)\\r\\nRN - 0 (RNA, Neoplasm)\\r\\nRN - CLE6G00625 (N-methyladenosine)\\r\\nRN - K72T3FS567 (Adenosine)\\r\\nSB - IM\\r\\nMH - Adenosine/*analogs & derivatives/chemistry\\r\\nMH - Animals\\r\\nMH - Biomarkers, Tumor/*genetics\\r\\nMH - *Gene Expression Regulation, Neoplastic\\r\\nMH - Humans\\r\\nMH - Liver Neoplasms/genetics/*pathology\\r\\nMH - RNA Processing, Post-Transcriptional\\r\\nMH - RNA, Neoplasm/*chemistry/*genetics\\r\\nPMC - PMC7047367\\r\\nOTO - NOTNLM\\r\\nOT - Epi-transcriptome\\r\\nOT - Hepatocellular carcinoma\\r\\nOT - N6-methyladenosine\\r\\nOT - RNA epigenetics\\r\\nCOIS- The authors declare that they have no competing interests.\\r\\nEDAT- 2020/03/01 06:00\\r\\nMHDA- 2021/02/02 06:00\\r\\nCRDT- 2020/03/01 06:00\\r\\nPHST- 2019/12/28 00:00 [received]\\r\\nPHST- 2020/02/24 00:00 [accepted]\\r\\nPHST- 2020/03/01 06:00 [entrez]\\r\\nPHST- 2020/03/01 06:00 [pubmed]\\r\\nPHST- 2021/02/02 06:00 [medline]\\r\\nAID - 10.1186/s12943-020-01172-y [pii]\\r\\nAID - 1172 [pii]\\r\\nAID - 10.1186/s12943-020-01172-y [doi]\\r\\nPST - epublish\\r\\nSO - Mol Cancer. 2020 Feb 28;19(1):44. doi: 10.1186/s12943-020-01172-y.\",\n",
              "  '@search.score': 1.1230905},\n",
              " {'filepath': 'pubmed-32045030.txt',\n",
              "  'content': \"PMID- 32045030\\r\\nOWN - NLM\\r\\nSTAT- MEDLINE\\r\\nDCOM- 20220107\\r\\nLR - 20220107\\r\\nIS - 1527-3350 (Electronic)\\r\\nIS - 0270-9139 (Linking)\\r\\nVI - 73 Suppl 1\\r\\nDP - 2021 Jan\\r\\nTI - Biomarkers for Hepatobiliary Cancers.\\r\\nPG - 115-127\\r\\nLID - 10.1002/hep.31175 [doi]\\r\\nAB - The clinical management of primary liver cancers such as hepatocellular carcinoma \\n (HCC) and cholangiocarcinoma (CCA) has significantly changed in the last 3\\xa0years. \\n The introduction of systemic therapies, including immune-based therapies and \\n biomarker-driven therapies, has significantly improved survival, particularly in \\n patients at advanced stages of disease. Survival is still poor, and projections \\n from the World Health Organization underscore the need to improve outcomes in \\n these patients. Biomarkers have emerged as powerful tools for the diagnosis, \\n prognosis, and prediction of treatment responses to improve patient \\n stratification and maximize clinical benefits. Recent advances in understanding \\n the molecular alterations of HCC have not yet translated into biomarkers. Some \\n reasons for the lack of progress are the absence of druggable mutations in the \\n majority of liver cancers and the significant heterogeneity of the disease. In \\n contrast, several therapeutic targets have been identified in CCA, and \\n biomarker-driven therapies are currently under evaluation in phase 2/3 clinical \\n trials. Here, we summarize the status on biomarker development for HCC and CCA.\\r\\nCI - © 2020 by the American Association for the Study of Liver Diseases.\\r\\nFAU - Nault, Jean-Charles\\r\\nAU - Nault JC\\r\\nAD - Centre de Recherche des Cordeliers, Functional Genomics of Solid Tumors \\n Laboratory, Sorbonne Université, Inserm, USPC, Université Paris Descartes, \\n Université Paris Diderot, Université Paris 13, Paris, France.\\r\\nAD - Liver Unit, Hôpital Jean Verdier, Hôpitaux Universitaires \\n Paris-Seine-Saint-Denis, Assistance-Publique Hôpitaux de Paris, Bondy, France.\\r\\nAD - Unité de Formation et de Recherche Santé Médecine et Biologie Humaine, Université \\n Paris 13, Paris, France.\\r\\nFAU - Villanueva, Augusto\\r\\nAU - Villanueva A\\r\\nAD - Division of Liver Diseases, Department of Medicine, Icahn School of Medicine at \\n Mount Sinai, Liver Cancer Program, Tisch Cancer Institute, New York, NY.\\r\\nAD - Division of Hematology and Medical Oncology, Department of Medicine, Icahn School \\n of Medicine at Mount Sinai, New York, NY.\\r\\nLA - eng\\r\\nPT - Journal Article\\r\\nPT - Research Support, N.I.H., Extramural\\r\\nPT - Research Support, U.S. Gov't, Non-P.H.S.\\r\\nPT - Review\\r\\nDEP - 20201129\\r\\nPL - United States\\r\\nTA - Hepatology\\r\\nJT - Hepatology (Baltimore, Md.)\\r\\nJID - 8302946\\r\\nRN - 0 (Biomarkers, Tumor)\\r\\nSB - IM\\r\\nMH - Bile Duct Neoplasms/diagnosis/*drug therapy/genetics\\r\\nMH - Biomarkers, Tumor\\r\\nMH - Carcinoma, Hepatocellular/diagnosis/*drug therapy/genetics\\r\\nMH - Cholangiocarcinoma/diagnosis/*drug therapy/genetics\\r\\nMH - Early Detection of Cancer\\r\\nMH - Humans\\r\\nMH - Liver Neoplasms/diagnosis/*drug therapy/genetics\\r\\nMH - Prognosis\\r\\nEDAT- 2020/02/12 06:00\\r\\nMHDA- 2022/01/08 06:00\\r\\nCRDT- 2020/02/12 06:00\\r\\nPHST- 2019/12/02 00:00 [received]\\r\\nPHST- 2020/01/28 00:00 [accepted]\\r\\nPHST- 2020/02/12 06:00 [pubmed]\\r\\nPHST- 2022/01/08 06:00 [medline]\\r\\nPHST- 2020/02/12 06:00 [entrez]\\r\\nAID - 10.1002/hep\",\n",
              "  '@search.score': 1.0495409},\n",
              " {'filepath': 'Cognitive Science Final Paper.docx',\n",
              "  'content': 'However, with the help of artificial intelligence, specifically 3D convolutional neural networks (3DCNNs) which is a subset of machine learning, it can sequence data and identify correlations quickly with the new data given. CNNs are used in deep learning for visual intelligence (analyzing and classifying visual imagery) (Hakim). \\n\\tA study performed by the Department of Computer Science and Information Engineering has focus on this intersection of 3D-CNN, and hand gesture recognition. They first inputted RGB (Red Green Blue), Multimodal data into the model, and used Finite State Machine (FSM) to limit the number of recognition classes, as smaller classes usually output more accuracy than bigger classes. Then they cleanup and modify the data by removing pixels in the background that are not needed, so they can just focus on the pixels of the hand. They then classify different hand gestures into 24 main isolated gestures (split into 11 static and 13 dynamic gestures) during the training and testing time-period (some of the gestures seen in Figure 1). After that, they collected gesture image data from 20 people who were in 5 different environments with different lighting, and used this data for training, and testing (with a 70% and 30% split respectively). All of the people who participated were right-handed (to improve consistency and reduce bias). The testing resulted in a >90% accuracy with the background/unnecessary pixels removed. They concluded that the use of 3DCNN could obtain spatio-temporal features in the gesture in a real time application (Hakim).\\n\\tAnother study has used a feature called Finger-Earth Movers Distance (FEMD) to understand finger level features (such as how far each of the fingers are from each other, finger position, along with what angle they are positioned at) (Hakim).\\n\\tConvolutional Neural Networks and Cognitive Science don’t just play a vital role in the medical field, but also in saving the planet. Another real-world application is real time analysis of the count of endangered species using CNNs and Drones. Currently, several animal species who are currently on the verge of extinction are severely impacted by poaching. Therefore, there have been efforts to halt poaching by installing hidden cameras and drones to identify/track poachers and make sure they don’t come near endangered animals. These cameras/drones contain CNNs which can process a real-time video and classify their data collection to identify hazards. The model processes the images in a cloud infrastructure which classifies the objects and the local node can determine whether the current frame has a hazard using an algorithm called Binarized Normed Gradients (Chalmers).\\n\\tAn experiment done by Chalmers, Fergus, Wich and Aday implemented cameras and used CNNs to identify certain objects. The dataset consisted of RGB & thermal pictures. There were only 2 classes: cars & rhinos. They inputted 350 pictures (of resolutions from 300 x 147 to 3840 x 2160 pixels) into each class for training. A drone captures the RGB images from above and the thermal/grey scale media is recorded from a camera on the ground. Then the CSV and Raster files (image data) is converted into TFRecord structure which is then inputted into training. The object detection process consists of 2 stages. The 1st stage involves Region Proposal Networks (RPN) which identifies, and extracts features from the layers which are selected. The 2nd stage fixes the bounding box tightly around the region of interest (to minimize the pixels not needed inside the box). This step improves the accuracy and speed of the CNN. The overall process is shown in Figure 2. Since they used transfer learning, the model was previously trained on millions of images but then refined by angling its focus towards rhino and car images (as more images trained leads to higher accuracy). When CNNs are trained on smaller datasets then that leads to a high amount of overfitting because the variance is too small. From this experiment they concluded that recent advances in neural networks, machine learning and cognitive science have proved to help species conservation in real time (Chalmers).\\n\\tThis section goes more in depth of how deep artificial intelligence neural networks work. Neural Networks contain many different layers of digital neurons (more layers corresponds to a deeper network). After the network is given input data, it can ‘self learn’ patterns without experts needing to apply correlations to every single pair. As long as it is given a list of labelled inputs along with the corresponding outputs, it can use that for training and then predict correlations on new real-time data not previously given. This is called supervised deep learning (Aakash).\\n\\tDeep learning has several promising advantages, but what about the potential limitations? The first limitation is that it is very resource-draining. It takes a high amount of computational power to train the models (with millions of images)',\n",
              "  '@search.score': 0.9617383},\n",
              " {'filepath': 'pubmed-7578390.txt',\n",
              "  'content': 'PMID- 7578390\\r\\nOWN - NLM\\r\\nSTAT- MEDLINE\\r\\nDCOM- 19951214\\r\\nLR - 20190920\\r\\nIS - 1040-8746 (Print)\\r\\nIS - 1040-8746 (Linking)\\r\\nVI - 7\\r\\nIP - 4\\r\\nDP - 1995 Jul\\r\\nTI - Primary liver cancers.\\r\\nPG - 387-96\\r\\nAB - A review of the recent literature about liver cancers shows the increasing role \\n of hepatitis C as a risk factor for the development of hepatocellular carcinoma. \\n A new discussion has arisen with regard to the association between the use of \\n oral contraceptives and hepatocellular carcinoma. Patients at risk for developing \\n liver cancer can be identified and should undergo thorough follow-up to detect \\n cancer at an early stage. Surgical resection remains the treatment of choice \\n whenever feasible. Because of disappointing results and limited availability of \\n donor organs, liver transplantation must be considered only in selected patients \\n with unresectable tumors. There are also arguments recommending transplantation \\n in early resectable hepatocellular carcinoma in patients with cirrhosis to cure \\n the underlying cancer-bearing disease. Improvement of survival can be expected by \\n promising multimodality treatment protocols that have been started during the \\n past few years. Further studies should also investigate the role of \\n immunosuppression in the treatment of tumor recurrence. For patients excluded \\n from resection and transplantation, the most effective treatment options are \\n transarterial chemoembolization and, in small lesions, percutaneous ethanol \\n injection therapy. In small tumors, combination therapy seems to be even more \\n appropriate.\\r\\nFAU - Weimann, A\\r\\nAU - Weimann A\\r\\nAD - Klinik für Abdominal- und Transplantationschirurgie, Medizinische Hochschule \\n Hannover, Germany.\\r\\nFAU - Oldhafer, K J\\r\\nAU - Oldhafer KJ\\r\\nFAU - Pichlmayr, R\\r\\nAU - Pichlmayr R\\r\\nLA - eng\\r\\nPT - Journal Article\\r\\nPT - Review\\r\\nPL - United States\\r\\nTA - Curr Opin Oncol\\r\\nJT - Current opinion in oncology\\r\\nJID - 9007265\\r\\nRN - 0 (Antineoplastic Agents)\\r\\nSB - IM\\r\\nMH - Antineoplastic Agents/therapeutic use\\r\\nMH - Bile Duct Neoplasms/diagnosis/epidemiology/therapy\\r\\nMH - Carcinoma, Hepatocellular/diagnosis/epidemiology\\r\\nMH - Chemoembolization, Therapeutic\\r\\nMH - Cholangiocarcinoma/epidemiology\\r\\nMH - Combined Modality Therapy\\r\\nMH - Female\\r\\nMH - Humans\\r\\nMH - *Liver Neoplasms/diagnosis/epidemiology/*therapy\\r\\nMH - Liver Transplantation\\r\\nMH - Male\\r\\nRF - 84\\r\\nEDAT- 1995/07/01 00:00\\r\\nMHDA- 1995/07/01 00:01\\r\\nCRDT- 1995/07/01 00:00\\r\\nPHST- 1995/07/01 00:00 [pubmed]\\r\\nPHST- 1995/07/01 00:01 [medline]\\r\\nPHST- 1995/07/01 00:00 [entrez]\\r\\nAID - 10.1097/00001622-199507000-00017 [doi]\\r\\nPST - ppublish\\r\\nSO - Curr Opin Oncol. 1995 Jul;7(4):387-96. doi: 10.1097/00001622-199507000-00017.',\n",
              "  '@search.score': 0.9507718},\n",
              " {'filepath': 'Written Report First Draft.docx',\n",
              "  'content': \"” EPA, Environmental Protection Agency, https://www.epa.gov/nutrientpollution/sources-and-solutions-agriculture.\\n· UC Davis: Nitrogen Fertilizer Loading to Groundwater Proposal to FREP, 2009. https://www.cdfa.ca.gov/is/docs/11-0301-SAHarter.pdf.\\n· US Department of Commerce, National Oceanic and Atmospheric Administration. “What Is Nutrient Pollution?” NOAA's National Ocean Service, 1 Sept. 2009, https://oceanservice.noaa.gov/facts/nutpollution.html.\\n· “USGS Current Water Data for the Nation.” USGS Current Water Data for the Nation, https://waterdata.usgs.gov/nwis/rt.\\n· Viets,, Frank G. “Water Quality in Relation to Farm Use of Fertilizer.” BioScience, vol. 21, no. 10, 1971, pp. 460–467., https://doi.org/10.2307/1295787.\\n· writer, About Arindom GhoshA professional, et al. “How Chemical Fertilizers Cause Water Pollution?” Conserve Energy Future, 28 July 2022, https://www.conserve-energy-future.com/how-chemical-fertilizers-cause-water-pollution.php.\\xa0\\n· Figures\\n*Experiment process: Soil collection, soil combined with fertilizer/compost, put in labeled mason jars, water added, straining the soil/water mixture, dipping test strips in solution, rinsing with water, final test strips from all solutions (left to right: water, control 1, control 2, compost 1, compost 2, fertilizer 1, fertilizer 2)\\n· Tables\\n*Mississippi River/Gulf of Mexico case study showcasing the process of a deadzone\\n*Water quality test strips used in experiment showcasing the 16 items that were tested.\\n*Fertilizer singles used containing phosphorus and nitrogen\\n*Results from experiment showcasing pH, nitrite, and nitrate differences among the various samples.\\n\\tSample\\n\\tTotal\\xa0\\nAlkalinity\\n\\tpH\\n\\tHardness\\n\\tCNOH3\\n\\tTotal\\xa0\\nCl\\n\\tFree\\nCl\\n\\tBr\\n\\tNO3\\n\\tNO2\\n\\tFe\\n\\tCr\\xa0\\n\\tPb\\n\\tCu\\n\\tHg\\n\\tF\\n\\tCarbonate\\xa0\\nRoot\\n\\tWater\\n\\t240\\n\\t6\\n\\t25\\n\\t0\\n\\t0\\n\\t0.5\\n\\t0.5\\n\\t10\\n\\t0\\n\\t0.2\\n\\t0\\n\\t0.015\\n\\t0\\n\\t0\\n\\t0\\n\\t40\\n\\tControl 1\\n\\t120\\n\\t6.5\\n\\t50\\n\\t0\\n\\t0.5\\n\\t0.5\\n\\t2\\n\\t30\\n\\t5\\n\\t1\\n\\t2\\n\\t0.03\\n\\t0.1\\n\\t0\\n\\t0\\n\\t\\n\\tControl 2\\n\\t40\\n\\t6.5\\n\\t50\\n\\t0\\n\\t0.5\\n\\t0.3\\n\\t0.3\\n\\t25\\n\\t0.5\\n\\t0.3\\n\\t0\\n\\t0.005\\n\\t0.1\\n\\t0\\n\\t2\\n\\t0\\n\\tCompost\\xa0\\n1\\n\\t200\\n\\t8.5\\n\\t425\\n\\t30-\\n50\\n\\t0.5\\n\\t0\\n\\t0\\n\\t4\\n\\t0.5\\n\\t1.0\\n\\t0\\n\\t0.015\\n\\t1.0\\n\\t0\\n\\t0\\n\\t15\\n\\tCompost 2\\n\\t120\\n\\t8\\n\\t425\\n\\t0\\n\\t0.5\\n\\t0\\n\\t0\\n\\t25\\n\\t1\\n\\t1\\n\\t0\\n\\t0.015\\n\\t0.1\\n\\t0\\n\\t10\\n\\t40\\n\\tFertilizer 1\\n\\t80\\n\\t7.5\\n\\t50\\n\\t10\\n\\t0.5\\n\\t0\\n\\t0\\n\\t10\\n\\t0\\n\\t0.1\\n\\t0\\n\\t0.005\\n\\t0.05\\n\\t0\\n\\t4\\n\\t0\\n\\tFertilizer 2\\n\\t40\\n\\t6.5\\n\\t25\\n\\t0\\n\\t0.5\\n\\t0\\n\\t0\\n\\t25\\n\\t0\\n\\t0.3\\n\\t0\\n\\t0.015\\n\\t0\",\n",
              "  '@search.score': 0.8859299},\n",
              " {'filepath': 'pubmed-32402299.txt',\n",
              "  'content': 'PMID- 32402299\\r\\nOWN - NLM\\r\\nSTAT- MEDLINE\\r\\nDCOM- 20200609\\r\\nLR - 20200609\\r\\nIS - 1558-3171 (Electronic)\\r\\nIS - 0039-6109 (Linking)\\r\\nVI - 100\\r\\nIP - 3\\r\\nDP - 2020 Jun\\r\\nTI - Primary Liver Cancers: Intrahepatic Cholangiocarcinoma and Hepatocellular \\n Carcinoma.\\r\\nPG - 535-549\\r\\nLID - S0039-6109(20)30026-8 [pii]\\r\\nLID - 10.1016/j.suc.2020.02.013 [doi]\\r\\nAB - The incidence of liver cancers has continued to increase over the past few \\n decades and mortality related to liver cancer has increased by more than 2% \\n annually since 2007. This article reviews the essential workup and treatment \\n options necessary for general surgeons as they treat patients with primary liver \\n cancers.\\r\\nCI - Copyright © 2020 Elsevier Inc. All rights reserved.\\r\\nFAU - Mejia, Juan C\\r\\nAU - Mejia JC\\r\\nAD - Providence Sacred Heart Medical Center, 101 West 8th Avenue, Suite 7050, Spokane, \\n WA 99204, USA. Electronic address: Juan.Mejia@providence.org.\\r\\nFAU - Pasko, Jennifer\\r\\nAU - Pasko J\\r\\nAD - Providence Sacred Heart Medical Center, 101 West 8th Avenue, Suite 7050, Spokane, \\n WA 99204, USA.\\r\\nLA - eng\\r\\nPT - Journal Article\\r\\nPT - Review\\r\\nDEP - 20200416\\r\\nPL - United States\\r\\nTA - Surg Clin North Am\\r\\nJT - The Surgical clinics of North America\\r\\nJID - 0074243\\r\\nSB - IM\\r\\nMH - Antineoplastic Combined Chemotherapy Protocols/therapeutic use\\r\\nMH - Bile Duct Neoplasms/diagnosis/pathology/*surgery\\r\\nMH - Carcinoma, Hepatocellular/diagnosis/pathology/*surgery\\r\\nMH - Cholangiocarcinoma/diagnosis/pathology/*surgery\\r\\nMH - Combined Modality Therapy\\r\\nMH - Endosonography\\r\\nMH - Humans\\r\\nMH - Laparoscopy\\r\\nMH - Liver Neoplasms/diagnosis/pathology/*surgery\\r\\nMH - Lymph Node Excision/methods\\r\\nMH - Lymphatic Metastasis/diagnosis/pathology\\r\\nMH - Margins of Excision\\r\\nMH - Neoplasm Staging\\r\\nMH - Positron Emission Tomography Computed Tomography\\r\\nMH - Positron-Emission Tomography\\r\\nMH - Randomized Controlled Trials as Topic\\r\\nMH - Sensitivity and Specificity\\r\\nMH - Tomography, X-Ray Computed\\r\\nOTO - NOTNLM\\r\\nOT - Cholangiocarcinoma\\r\\nOT - Cirrhosis\\r\\nOT - Hepatocellular carcinoma\\r\\nOT - Intrahepatic cholangiocarcinoma\\r\\nOT - Liver resection\\r\\nOT - Primary liver cancers\\r\\nCOIS- Disclosure No disclosures.\\r\\nEDAT- 2020/05/14 06:00\\r\\nMHDA- 2020/06/10 06:00\\r\\nCRDT- 2020/05/14 06:00\\r\\nPHST- 2020/05/14 06:00 [entrez]\\r\\nPHST- 2020/05/14 06:00 [pubmed]\\r\\nPHST- 2020/06/10 06:00 [medline]\\r\\nAID - S0039-6109(20)30026-8 [pii]\\r\\nAID - 10.1016/j.suc.2020.02.013 [doi]\\r\\nPST - ppublish\\r\\nSO - Surg Clin North Am. 2020 Jun;100(3):535-549. doi: 10.1016/j.suc.2020.02.013. Epub \\n 2020 Apr 16.',\n",
              "  '@search.score': 0.8788067},\n",
              " {'filepath': 'Saumya_Soni_Latest_Resume.pdf',\n",
              "  'content': 'Presented in the school assembly garner \\ndonations; collected $2900, equivalates to 35,000 trees planted \\n \\nCo-founder, “Leaders Behind Screens” initiative by Ampowering Non-profit Organization (Summer 2021) \\n▪ Co-created and co-lead a customized education plan for 8 weeks; taught Environmental Sustainability \\n▪ Received “Exceptional Youth Leadership” Award from AMpowering \\nLeader, National Outreach Team of ‘The Tomorrow Project’ (March 2021 – Present) \\n▪ Formalize workshops about climate change/ environmental topics and present them to elementary/middle schools \\n▪ Reach out to different organizations to collaborate, write blogs, and speak about sustainability in different conferences \\nE X T R A C U R R I C U L A R A C T I V I T I E S \\nSinger and Teaching Assistant, Swaranjali School of Music (2014-present) \\n▪ Earned Sr. Diploma in Indian (Hindustani) Classical music, learned singing for last 10+ years and assisted in teaching \\n▪ One of the 3 winners for CRY Singing Superstar in Seattle (2019) & Finalist in nationwide competition (2020 & 2021) \\n▪ Performed, including solo performances in 5+ shows with audiences of up to 8,000 people \\nSecond Place Varsity Debate Winner (2 times), Capitol Debate (2017 & 2018): Debated topics including: Does Artificial \\nIntelligence do more good than harm? Should healthcare be universal? \\nRunner, Varsity Track & Field at EPS (2019) and Speed Unlimited Track Club (2017-2019). Participated in Junior Olympics \\nEngineering Notebook Manager, Robotics, Eastside Prep. School (2018, 2019) \\nDancer, PCIPA Bollywood Dance (2013-2019), Scuba Diver (10 countries) and World traveller (33 countries) \\nhttps://www.youtube.com/playlist?list=PL-cSddcdvlup3eIHc8QcQ-yYjispKN9P7\\nhttps://www.youtube.com/playlist?list=PL-cSddcdvlup3eIHc8QcQ-yYjispKN9P7\\nhttps://www.youtube.com/playlist?list=PL-cSddcdvlurEXOfBAUB0c7F7svt9-WQg\\nhttps://ampowering.org/ampowering-team/',\n",
              "  '@search.score': 0.8145475},\n",
              " {'filepath': 'Written Report First Draft.docx',\n",
              "  'content': '2\\nLA12 - Lab Section 101\\nGroup Members: Alice Brookston, Caitlin Hubbs, Saumya Soni, Vicky Hao\\nImpacts of Agricultural Fertilizer on Water Quality to Nearby Sources of Water\\n· Abstract\\n· The Green Revolution in the 1950s and 60s revolutionized the world of agriculture through mechanized and synthetic farming, and at the same time permanently altered and degraded the environment, an event which has changed the physical ecology and composition of both soil and water, an impact that was only furthered by the usage of synthetic fertilizers. Fertilizer is essentially a chemical mixture added to soil to help replenish missing soils, namely nitrogen and phosphorus which most plants cannot create themselves but need to properly grow. Caused by years of soil degradation, most large farms do not hold as much water as they used to, and because of such, runoff is a regular occurrence and source of stress for local waterways which take on said burden. Seeing as most large-scale agricultural productions use synthetic fertilizers to increase yield, when said water runs off, it brings with it the contents of said fertilizer-rich soil, depositing them into local waterways, altering its chemical composition and causing damage to the organisms at hand. Recent studies have made it clear that this has had detrimental effects and can kill off organisms, altering the food chain, and making the water body much less viable for life. To display this, we performed our own experiment that displayed this idea in a smaller, controlled circumstance, while also researching real events where these effects have been seen to help further display this principle that synthetic fertilizer negatively affects water quality. Our own experiment unfortunately displayed inconclusive results, likely from our flaws in performing such, in which the data was too similar to support our hypothesis, although from online case studies and research, our theory still rings true. Overall, it still appears clear that synthetic fertilizer causes detrimental effects to water bodies because of the runoff of its chemicals.\\n· Introduction\\n· The purpose of this research project is to test how runoff from farms that use fertilizer affect water quality. Water quality can be measured in the chemical, physical, and biological contents of water. For our project, we will be looking at the chemical content of the water. Because there are so many minerals in water, we will examine the minerals that are more present in fertilizer. Fertilizers can be natural or synthetic, and are used to supply the soil with chemical elements to improve plant growth. Natural fertilizer is made of organic materials that supply nutrients (mainly nitrogen) to the soil, and can be used less often than synthetic fertilizer, which supplies these nutrients to the soil through a mineral form, which is inorganic. They contain nitrogen, phosphorus, and/or ammonia. Mixed fertilizers contain more than one of these elements. However, nutrient levels in soil are often higher than levels in crops, creating more extreme problems with the runoff from farms that use fertilizers. The runoff from these farms is water being carried off the farms from the surface, but the fertilizer makes the water heavily contaminated with nutrients. In addition, synthetic fertilizers are put into the soil already in mineral form, so the vast amount of nutrients not absorbed by the plants are carried into streams, polluting the water. Once this water has a surplus of nutrients, it causes algae growth, creating “dead zones”: areas in water where the oxygen supply has been cut off, which kills the animals in the water. Our research project will test the water quality of the runoff water of different fertilizers to determine which is best for the runoff and therefore surrounding environment. The results can be applied to agriculture and which fertilizer least affects the water quality of runoff.\\n· Methods\\n· Library/Internet Research: Online, individual case studies of fertilizer runoff impacts were researched in order to display how it can have negative effects on nearby water bodies because of its excess amount of nitrogen and phosphorus, changing the water’s chemical composition and affecting its ecology.\\n· Experiment: For the experiment, six bottles were filled with soil and water, two with nothing, two with compost, and two with fertilizer. These bottles were filled and left to sit for four days in order to saturate the water with the soil’s chemical compounds. Next, these bottles were strained, leaving the water behind which was tested for water quality, with a specific focus on pH and nitrogen. For testing, water quality test strips were dipped in a tube of the solution for two seconds, let hang to dry for 30 seconds, then rinsed off to make the results more clear. Additionally, the water used at the beginning was tested to act as a baseline. From this, data was collected based on the test strips dipped in each container.\\xa0\\n· Some potential flaws in the experiment:\\n1. The compost used contained chicken feces which has high levels of nitrogen, similar to fertilizer.\\n1',\n",
              "  '@search.score': 0.8102636},\n",
              " {'filepath': 'Written Report First Draft.docx',\n",
              "  'content': \"· Water Quality Standards\\nHigh nutrient content in water can lead to low oxygen levels, creating an imbalance. Water quality can be assessed through measuring the chemical, physical, and biological content in water. Good quality water should have low levels of pollutants or contaminants, but rich in nutrients and oxygen. Through our water quality test that measured chemicals that are most prevalent in fertilizers, we realized that there were so many chemicals that others can seem insignificant. Results from the strip tests showed that the water’s pH level ranged from 6.0-8.5 (the lower the pH, the more corrosive the water is); the nitrates: >10 mg/l as NO3-N, >45 mg/l as NO3; lead >15 ug; Chloride >250 mg/l; Fluoride: 0.7-1.2 mg/l; and Iron: >0.3 mg/l.\\n· Conclusions\\n· From our research and experiment we can conclude that fertilizers cause negative impacts to their nearby aquatic environments, although the scope of how severe the damages are depends on how much fertilizer was used on how large a piece of land, along with how large the nearby waterbody is. Overall though, the nutrients in the fertilizer have a likely chance to minimize the water’s organism diversity, cause dead zones, and overall dampen the well-being of the ecology.\\n· References\\xa0\\n· Advanced Solutions International, Inc. “What Are Fertilizers?” What Are Fertilizers, https://www.fertilizer.org/Public/About_fertilizers/What_Are_Fertilizers/Public/About_Fertilizers/What_are_Fertilizers.aspx.\\n· “Agricultural Contaminants Active.” Agricultural Contaminants | U.S. Geological Survey, https://www.usgs.gov/mission-areas/water-resources/science/agricultural-contaminants.\\n· Bijay-Singh, and Eric Craswell. “Fertilizers and Nitrate Pollution of Surface and Ground Water: An Increasingly Pervasive Global Problem - SN Applied Sciences.” SpringerLink, Springer International Publishing, 31 Mar. 2021, https://link.springer.com/article/10.1007/s42452-021-04521-8.\\n· Charles, Dan. “Iowa's Largest City Sues over Farm Fertilizer Runoff in Rivers.” NPR, NPR, 12 Jan. 2015, https://www.npr.org/sections/thesalt/2015/01/12/376139473/iowas-largest-city-sues-over-farm-fertilizer-runoff-in-rivers.\\n· “Fertilizer.” Encyclopædia Britannica, Encyclopædia Britannica, Inc., https://www.britannica.com/topic/fertilizer.\\n· González-Rivas, David A., et al. “Estimating Nitrogen Runoff from Agriculture to Coastal Zones by a Rapid Gis and Remote Sensing-Based Method for a Case Study from the Irrigation District Río Mayo, Gulf of California, México.” Frontiers, Frontiers, 1 Jan. 1AD, https://www.frontiersin.org/articles/10.3389/fmars.2020.00316/full.\\n· Lead Author Lead Author: Reviewed by Tom Scherer, and Lead Author: Reviewed by Tom Scherer. “Drinking Water Quality: Testing and Interpreting Your Results.” NDSU Agriculture and Extension, 24 Feb. 2022, https://www.ndsu.edu/agriculture/extension/publications/drinking-water-quality-testing-and-interpreting-your-results.\\n· Michigan Farmers Using Too Much Fertilizer, Hurting Water Quality Efforts. https://www.bridgemi.com/michigan-environment-watch/michigan-farmers-using-too-much-fertilizer-hurting-water-quality-efforts.\\n· “Mississippi River Case Study.” Student Materials, 9 May 2022, https://serc.carleton.edu/integrate/teaching_materials/food_supply/student_materials/1113.\\n· “Nitrogen and Water Completed.” Nitrogen and Water | U.S. Geological Survey, https://www.usgs.gov/special-topics/water-science-school/science/nitrogen-and-water.\\n· “The Sources and Solutions: Agriculture\",\n",
              "  '@search.score': 0.72675514},\n",
              " {'filepath': 'pubmed-31378681.txt',\n",
              "  'content': 'FAU - Xie, Fubo\\r\\nAU - Xie F\\r\\nAD - Shanghai ChemPartner Co., Ltd., Shanghai 201203, China.\\r\\nFAU - Gu, Ying\\r\\nAU - Gu Y\\r\\nAD - Shanghai ChemPartner Co., Ltd., Shanghai 201203, China.\\r\\nFAU - Zou, Keke\\r\\nAU - Zou K\\r\\nAD - CAS Key Laboratory of Computational Biology, CAS-MPG Partner Institute for \\n Computational Biology, Shanghai Institute of Nutrition and Health, Shanghai \\n Institutes for Biological Sciences, University of Chinese Academy of Sciences, \\n Chinese Academy of Sciences, Shanghai 200031, China.\\r\\nFAU - Li, Chao\\r\\nAU - Li C\\r\\nAD - CAS Key Laboratory of Computational Biology, CAS-MPG Partner Institute for \\n Computational Biology, Shanghai Institute of Nutrition and Health, Shanghai \\n Institutes for Biological Sciences, University of Chinese Academy of Sciences, \\n Chinese Academy of Sciences, Shanghai 200031, China.\\r\\nFAU - Li, Chun\\r\\nAU - Li C\\r\\nAD - State Key Laboratory of Cell Biology, CAS Center for Excellence in Molecular Cell \\n Science, Shanghai Institute of Biochemistry and Cell Biology, University of \\n Chinese Academy of Sciences, Chinese Academy of Sciences, Shanghai 200031, China.\\r\\nFAU - Wang, Chenhua\\r\\nAU - Wang C\\r\\nAD - State Key Laboratory of Cell Biology, CAS Center for Excellence in Molecular Cell \\n Science, Shanghai Institute of Biochemistry and Cell Biology, University of \\n Chinese Academy of Sciences, Chinese Academy of Sciences, Shanghai 200031, China.\\r\\nFAU - Cen, Jin\\r\\nAU - Cen J\\r\\nAD - State Key Laboratory of Cell Biology, CAS Center for Excellence in Molecular Cell \\n Science, Shanghai Institute of Biochemistry and Cell Biology, University of \\n Chinese Academy of Sciences, Chinese Academy of Sciences, Shanghai 200031, China.\\r\\nFAU - Chen, Xiaotao\\r\\nAU - Chen X\\r\\nAD - State Key Laboratory of Cell Biology, CAS Center for Excellence in Molecular Cell \\n Science, Shanghai Institute of Biochemistry and Cell Biology, University of \\n Chinese Academy of Sciences, Chinese Academy of Sciences, Shanghai 200031, China.\\r\\nFAU - Shu, Yajing\\r\\nAU - Shu Y\\r\\nAD - State Key Laboratory of Cell Biology, CAS Center for Excellence in Molecular Cell \\n Science, Shanghai Institute of Biochemistry and Cell Biology, University of \\n Chinese Academy of Sciences, Chinese Academy of Sciences, Shanghai 200031, China.\\r\\nFAU - Zhang, Zhao\\r\\nAU - Zhang Z\\r\\nAD - State Key Laboratory of Cell Biology, CAS Center for Excellence in Molecular Cell \\n Science, Shanghai Institute of Biochemistry and Cell Biology, University of \\n Chinese Academy of Sciences, Chinese Academy of Sciences, Shanghai 200031, China.\\r\\nFAU - Sun, Lulu\\r\\nAU - Sun L\\r\\nAD - State Key Laboratory of Cell Biology, CAS Center for Excellence in Molecular Cell \\n Science, Shanghai Institute of Biochemistry and Cell Biology, University of \\n Chinese Academy of Sciences, Chinese Academy of Sciences, Shanghai 200031, China.\\r\\nFAU - Min, Lihua\\r\\nAU - Min L\\r\\nAD - State Key Laboratory of Cell Biology, CAS Center for Excellence in Molecular Cell \\n Science, Shanghai Institute of Biochemistry and Cell Biology, University of \\n Chinese Academy of Sciences, Chinese Academy of Sciences, Shanghai 200031, China.\\r\\nFAU - Fu, Yong\\r\\nAU - Fu Y\\r\\nAD - Fifth Department of Hepatic Surgery, Eastern Hepatobiliary Surgery Hospital, \\n Second Military Medical University, Shanghai 200438, China.\\r\\nFAU - Huang, Xiaowu\\r\\nAU - Huang X\\r\\nAD - Liver Cancer Institute, Zhongshan Hospital, Fudan University, Key Laboratory of \\n Carcinogenesis and Cancer Invasion, Fudan University, Shanghai 200032, China.\\r\\nFAU - Lv, Hui\\r\\nAU - Lv H\\r\\nAD - SJTU-Yale Joint Center for Biostatistics, Department of Bioinformatics and \\n Biostatistics, Shanghai Jiaotong University, Shanghai 200240, China.\\r\\nFAU - Zhou, He\\r\\nAU - Zhou H\\r\\nAD - Shanghai ChemPartner Co., Ltd., Shanghai 201203, China.\\r\\nFAU - Ji, Yuan\\r\\nAU - Ji Y\\r\\nAD - Department of Pathology, Zhongshan Hospital, Fudan University, Shanghai 200032, \\n China',\n",
              "  '@search.score': 0.7245507},\n",
              " {'filepath': 'Cognitive Science Final Paper.docx',\n",
              "  'content': 'Soni 2\\nSaumya Soni\\nDr. Adam Croom\\nCognitive Science 1B\\n12/19/2022\\nHow Cognitive Science with AI is revolutionizing the Medical and Environmental Field\\nPart 1: It’s amazing how fast the cognitive science field is evolving and progressing. A person with motor disabilities wouldn’t have access to a solution about 50 years ago. However, in the current era of cognitive science, brilliant scientists have invented a revolutionary method to aid people with motor disabilities: Brain computer Interfaces (BCI). BCI’s are able to restore function and facilitate rehabilitation in individual who have to face trauma and/or disease. This technology can act as a substitute/restoration for muscles by enabling the individual to interact with their surroundings. This can be done by the BCI collecting electrophysiological signals from the brain to control a neuroprosthetic arm or understand behavior. It also acts to restore motor function by prompting the individual to do certain tasks so different brain signals are activated (JR Daly). \\n\\tResearch on this technology has come a long way, and there are four main aspects which aided its rapid growth. The first is the study of animals’ and human beings’ brains to see how specific brain signal’s function and what each of their uses/characteristics are. The second aspect is the understanding that activity-dependent plasticity happens in all parts of the central nervous system and therefore has a big influence on predicting the impacts of trauma and disease. The third aspect is the increased access to supporting technology which can aid in collecting and analyzing brain signals in real-time. The fourth aspect is the rise in societal demand for solutions to individuals who have to face tough mental and physical disabilities (JR Daly).\\n\\tThe main use of BCI’s is to substitute or improve neuro-muscular movement by analyzing the person’s intent in real-time. The process starts with the BCI collecting electrophysiological signals on the scalp of the individual and then translating the signals into a goal for what the person wants to do, and then it transfers over to the device commands to output the action. Different electrophysiological signals which are acquainted in BCI systems record data at different parts in the brain, some areas being more risk likely than others, while some areas containing a higher resolution than others (JR Daly).\\n\\tTo understand further about the process of how BCI fully translates brain signals into actions, its important to break down the process into steps. The first step is feature extraction; This is what collects the signals and breaks it down into simple observations/measures (e.g. the height of different energy potentials or the firing speed of neurons). The second step is to understand the signals by translating the signal into commands which are done with a translation algorithm. These commands correspond the signal to a output which the person desires to accomplish (such as moving their arm left or right, typing a certain letter, moving the computer cursor to select something, etc.) (JR Daly).\\n\\tCurrently, there are 3 types of BCI technologies which are being used: EEG-based BCIs, ECoG-based BCIs, and intracortical BCIs. EEG-based BCIs are recorded on the centroparietal cortex area of the scalp, where the software will show a range of selections to the user, and the user will spark a P300 potential at the desired selection. Then the BCI can recognize this sudden neuron spark and process the word/action. An ECoG BCI can use sensorimotor rhythms to identify the movement the user wants to make on the cursor. ECoGs are found to work faster and more efficiently than EEGs, as ECoGs have an even higher resolution and higher range of frequencies to identify. Intracortical BCIs includes multi-electrode arrays which can enable the cursor to move in more than 2 dimensions. They are usually implanted in the user’s motor cortex to collect the neuronal action potentials. With this, the software can correlate the target direction with the firing rate of 1 neuron (JR Daly).\\n\\tFor the BCIs to effectively understand the user’s brain signals, the user has to learn to use BCIs as well. The plasticity in the user’s neurons aid in learning new information along with acquiring new abilities. The BCI’s collect data about the motor and cognitive skills that correlate to the action the person wants to accomplish. The BCI user can use motor imagery to change the amplitude of rhythms, and over time the usage of a BCI starts to become automatic, and instead of physical muscle movement, the user controls the cursor through neuron signals (JR Daly)',\n",
              "  '@search.score': 0.63890904},\n",
              " {'filepath': 'pubmed-26187874.txt',\n",
              "  'content': 'PMID- 26187874\\r\\nOWN - NLM\\r\\nSTAT- MEDLINE\\r\\nDCOM- 20160517\\r\\nLR - 20181113\\r\\nIS - 2157-1422 (Electronic)\\r\\nIS - 2157-1422 (Linking)\\r\\nVI - 5\\r\\nIP - 9\\r\\nDP - 2015 Jul 17\\r\\nTI - Treatment of Liver Cancer.\\r\\nPG - a021535\\r\\nLID - 10.1101/cshperspect.a021535 [doi]\\r\\nLID - a021535\\r\\nAB - Primary liver cancer, mostly hepatocellular carcinoma, remains a \\n difficult-to-treat cancer. Incidence of liver cancer varies geographically and \\n parallels with the geographic prevalence of viral hepatitis. A number of staging \\n systems have been developed, reflecting the heterogeneity of primary liver \\n cancer, regional preferences, and regional variations in resectability or \\n transplant eligibility. Multimodality treatments are available for this \\n heterogeneous malignancy, and there are variations in the management \\n recommendations for liver cancers across specialties and geographic regions. \\n Novel treatment strategies have merged with the advance of new treatment \\n modalities. This work focuses on reviewing the incidence, staging, and treatment \\n of liver cancer.\\r\\nCI - Copyright © 2015 Cold Spring Harbor Laboratory Press; all rights reserved.\\r\\nFAU - Liu, Chun-Yu\\r\\nAU - Liu CY\\r\\nAD - Department of Medicine, Division of Hematology and Oncology, Taipei Veterans \\n General Hospital, Taipei 112, Taiwan School of Medicine, National Yang-Ming \\n University, Taipei 112, Taiwan Institute of Biopharmaceutical Sciences, National \\n Yang-Ming University, Taipei 112, Taiwan.\\r\\nFAU - Chen, Kuen-Feng\\r\\nAU - Chen KF\\r\\nAD - Department of Medical Research, National Taiwan University College of Medicine, \\n Taipei 112, Taiwan National Center of Excellence for Clinical Trial and Research, \\n National Taiwan University College of Medicine, Taipei 112, Taiwan.\\r\\nFAU - Chen, Pei-Jer\\r\\nAU - Chen PJ\\r\\nAD - Department of Medical Research, National Taiwan University College of Medicine, \\n Taipei 112, Taiwan National Center of Excellence for Clinical Trial and Research, \\n National Taiwan University College of Medicine, Taipei 112, Taiwan Graduate \\n Institute of Molecular Medicine, National Taiwan University Hospital and National \\n Taiwan University College of Medicine, Taipei 112, Taiwan.\\r\\nLA - eng\\r\\nPT - Journal Article\\r\\nPT - Review\\r\\nDEP - 20150717\\r\\nPL - United States\\r\\nTA - Cold Spring Harb Perspect Med\\r\\nJT - Cold Spring Harbor perspectives in medicine\\r\\nJID - 101571139\\r\\nRN - 0 (Antineoplastic Agents)\\r\\nSB - IM\\r\\nMH - Antineoplastic Agents/therapeutic use\\r\\nMH - Carcinoma, Hepatocellular/pathology/*therapy\\r\\nMH - Catheter Ablation/methods\\r\\nMH - Combined Modality Therapy\\r\\nMH - Hepatectomy/methods\\r\\nMH - Humans\\r\\nMH - Immunotherapy/methods\\r\\nMH - Liver Neoplasms/pathology/*therapy\\r\\nMH - Liver Transplantation/methods\\r\\nMH - Molecular Targeted Therapy/methods\\r\\nMH - Neoplasm Staging\\r\\nMH - Oncolytic Virotherapy/methods\\r\\nPMC - PMC4561392\\r\\nEDAT- 2015/07/19 06:00\\r\\nMHDA- 2016/05/18 06:00\\r\\nCRDT- 2015/07/19 06:00\\r\\nPHST- 2015/07/19 06:00 [entrez]\\r\\nPHST- 2015/07/19 06:00 [pubmed]\\r\\nPHST- 2016/05/18 06:00 [medline]\\r\\nAID - cshperspect.a021535 [pii]\\r\\nAID - a021535 [pii]\\r\\nAID - 10.1101/cshperspect.a021535 [doi]\\r\\nPST - epublish\\r\\nSO - Cold Spring Harb Perspect Med. 2015 Jul 17;5(9):a021535. doi: \\n 10.1101/cshperspect.a021535.',\n",
              "  '@search.score': 0.63370943},\n",
              " {'filepath': 'pubmed-31666728.txt',\n",
              "  'content': \"PMID- 31666728\\r\\nOWN - NLM\\r\\nSTAT- MEDLINE\\r\\nDCOM- 20200210\\r\\nLR - 20200826\\r\\nIS - 1759-5053 (Electronic)\\r\\nIS - 1759-5045 (Linking)\\r\\nVI - 16\\r\\nIP - 12\\r\\nDP - 2019 Dec\\r\\nTI - Metabolic rearrangements in primary liver cancers: cause and consequences.\\r\\nPG - 748-766\\r\\nLID - 10.1038/s41575-019-0217-8 [doi]\\r\\nAB - Primary liver cancer (PLC) is the fourth most frequent cause of cancer-related \\n death. The high mortality rates arise from late diagnosis and the limited \\n accuracy of diagnostic and prognostic biomarkers. The liver is a major regulator, \\n orchestrating the clearance of toxins, balancing glucose, lipid and amino acid \\n uptake, managing whole-body metabolism and maintaining metabolic homeostasis. \\n Tumour onset and progression is frequently accompanied by rearrangements of \\n metabolic pathways, leading to dysregulation of metabolism. The limitation of \\n current therapies targeting PLCs, such as hepatocellular carcinoma and \\n cholangiocarcinoma, points towards the importance of deciphering this metabolic \\n complexity. In this Review, we discuss the role of metabolic liver disruptions \\n and the implications of these processes in PLCs, emphasizing their clinical \\n relevance and value in early diagnosis and prognosis and as putative therapeutic \\n targets. We also describe system biology approaches able to reconstruct the \\n metabolic complexity of liver diseases. We also discuss whether metabolic \\n rearrangements are a cause or consequence of PLCs, emphasizing the opportunity to \\n clinically exploit the rewired metabolism. In line with this idea, we discuss \\n circulating metabolites as promising biomarkers for PLCs.\\r\\nFAU - Satriano, Letizia\\r\\nAU - Satriano L\\r\\nAD - Biotech Research and Innovation Centre (BRIC) Department of Health and Medical \\n Sciences, University of Copenhagen, Copenhagen, Denmark.\\r\\nFAU - Lewinska, Monika\\r\\nAU - Lewinska M\\r\\nAUID- ORCID: 0000-0002-5275-992X\\r\\nAD - Biotech Research and Innovation Centre (BRIC) Department of Health and Medical \\n Sciences, University of Copenhagen, Copenhagen, Denmark.\\r\\nFAU - Rodrigues, Pedro M\\r\\nAU - Rodrigues PM\\r\\nAUID- ORCID: 0000-0001-6193-7436\\r\\nAD - Biodonostia Health Research Institute, Donostia University Hospital, San \\n Sebastian, Spain.\\r\\nFAU - Banales, Jesus M\\r\\nAU - Banales JM\\r\\nAUID- ORCID: 0000-0002-5224-2373\\r\\nAD - Biodonostia Health Research Institute, Donostia University Hospital, San \\n Sebastian, Spain.\\r\\nAD - National Institute for the Study of Liver and Gastrointestinal Diseases \\n (CIBERehd), Carlos III National Institute of Health, Madrid, Spain.\\r\\nAD - IKERBASQUE, Basque Foundation for Science, Bilbao, Spain.\\r\\nFAU - Andersen, Jesper B\\r\\nAU - Andersen JB\\r\\nAUID- ORCID: 0000-0003-1760-5244\\r\\nAD - Biotech Research and Innovation Centre (BRIC) Department of Health and Medical \\n Sciences, University of Copenhagen, Copenhagen, Denmark. \\n jesper.andersen@bric.ku.dk.\\r\\nLA - eng\\r\\nPT - Journal Article\\r\\nPT - Research Support, Non-U.S. Gov't\\r\\nPT - Review\\r\\nDEP - 20191030\\r\\nPL - England\\r\\nTA - Nat Rev Gastroenterol Hepatol\\r\\nJT - Nature reviews\",\n",
              "  '@search.score': 0.5750315},\n",
              " {'filepath': 'pubmed-36158262.txt',\n",
              "  'content': \"PMID- 36158262\\r\\nOWN - NLM\\r\\nSTAT- MEDLINE\\r\\nDCOM- 20220928\\r\\nLR - 20220928\\r\\nIS - 2219-2840 (Electronic)\\r\\nIS - 1007-9327 (Print)\\r\\nIS - 1007-9327 (Linking)\\r\\nVI - 28\\r\\nIP - 27\\r\\nDP - 2022 Jul 21\\r\\nTI - Artificial intelligence in liver ultrasound.\\r\\nPG - 3398-3409\\r\\nLID - 10.3748/wjg.v28.i27.3398 [doi]\\r\\nAB - Artificial intelligence (AI) is playing an increasingly important role in \\n medicine, especially in the field of medical imaging. It can be used to diagnose \\n diseases and predict certain statuses and possible events that may happen. \\n Recently, more and more studies have confirmed the value of AI based on \\n ultrasound in the evaluation of diffuse liver diseases and focal liver lesions. \\n It can assess the severity of liver fibrosis and nonalcoholic fatty liver, \\n differentially diagnose benign and malignant liver lesions, distinguish primary \\n from secondary liver cancers, predict the curative effect of liver cancer \\n treatment and recurrence after treatment, and predict microvascular invasion in \\n hepatocellular carcinoma. The findings from these studies have great clinical \\n application potential in the near future. The purpose of this review is to \\n comprehensively introduce the current status and future perspectives of AI in \\n liver ultrasound.\\r\\nCI - ©The Author(s) 2022. Published by Baishideng Publishing Group Inc. All rights \\n reserved.\\r\\nFAU - Cao, Liu-Liu\\r\\nAU - Cao LL\\r\\nAD - Department of Medical Ultrasound, The Second Hospital of Anhui Medical \\n University, Hefei 230601, Anhui Province, China.\\r\\nFAU - Peng, Mei\\r\\nAU - Peng M\\r\\nAD - Department of Medical Ultrasound, The Second Hospital of Anhui Medical \\n University, Hefei 230601, Anhui Province, China.\\r\\nFAU - Xie, Xiang\\r\\nAU - Xie X\\r\\nAD - Department of Medical Ultrasound, The Second Hospital of Anhui Medical \\n University, Hefei 230601, Anhui Province, China.\\r\\nFAU - Chen, Gong-Quan\\r\\nAU - Chen GQ\\r\\nAD - Department of Medical Ultrasound, Minda Hospital of Hubei Minzu University, Enshi \\n 445000, Hubei Province, China.\\r\\nFAU - Huang, Shu-Yan\\r\\nAU - Huang SY\\r\\nAD - Department of Medical Ultrasound, The First People's Hospital of Huaihua, Huaihua \\n 418000, Hunan Province, China.\\r\\nFAU - Wang, Jia-Yu\\r\\nAU - Wang JY\\r\\nAD - Department of Medical Ultrasound, Tongji Hospital of Tongji Medical College, \\n Huazhong University of Science and Technology, Wuhan 430030, Hubei Province, \\n China.\\r\\nFAU - Jiang, Fan\\r\\nAU - Jiang F\\r\\nAD - Department of Medical Ultrasound, The Second Hospital of Anhui Medical \\n University, Hefei 230601, Anhui Province, China. ahultrasound2005@126.com.\\r\\nFAU - Cui, Xin-Wu\\r\\nAU - Cui XW\\r\\nAD - Department of Medical Ultrasound, Tongji Hospital of Tongji Medical College, \\n Huazhong University of Science and Technology, Wuhan 430030, Hubei Province, \\n China.\\r\\nFAU - Dietrich, Christoph F\\r\\nAU - Dietrich CF\\r\\nAD - Department Allgemeine Innere Medizin, Kliniken Hirslanden Beau Site, Salem und \\n Permanence, Bern 3626, Switzerland\",\n",
              "  '@search.score': 0.5269685},\n",
              " {'filepath': 'Cognitive Science Final Paper.docx',\n",
              "  'content': 'BCI’s have a wide range of applications, as they can offer aid in tasks as simple as answering yes or no questions, clicking switches and buttons, along with offering aid in more complex tasks such as operating a prosthetic hand (with multidimensional movement), a computer, etc. To restore the specific functions in the central nervous system, its important to identify and target activity-dependent brain plasticity. The ravishing technology can also provide a therapeutic application. It has been observed to lower the rate at which individuals with epilepsy have seizures, along with lowering anxiety rates. It can reopen volitional motor control, which is usually damaged by disease or trauma. The way BCIs are lowering the rate of seizures is by training the patients to exhibit normal brain behavior. With more normal brain activity, the central nervous system will restore its normal functions. Studies done on animals and humans have proved this method effective. In one study, animals were seen to show motor recovery after a stroke due to structural changes, like neurite outgrowth, increased excitability, and increased usage of growth-promoting genes (JR Daly). \\n\\tThe future of BCI will depend on further research into the feasibility/convenience along with the safety of specific sub-technologies under BCIs. Hopefully the correlation between brain activity and effective motor function will become stronger in the coming years, so the wide-range of applications for BCI can extend to a larger scale. With more research, BCIs can focus its applications on improving quality of life, desires, mood and productivity. The availability of BCIs isn’t as widespread currently as it is expensive and requires a lengthy amount of helping equipment, and technical support from researchers to keep it intact. To increase its availability, it needs to evolve so it is easy to operate and maintain. \\n\\tPart 2: In an era filled with technological innovations, BCI’s are also crossing paths with other revolutionary inventions, one being artificial intelligence. Artificial Intelligence (AI) can speed up the analysis and decode neural activity in a more effective way. Studies have shown that smart BCI’s, powered by AI, improve the quality of life of individuals who face paralysis. To understand further on the intersection between BCIs and Artificial Intelligence, its first important to understand what artificial intelligence is. Artificial Intelligence is a computer program/algorithm which can mimic human intelligence and apply it in a range of applications such as speech recognition, translation, analysis, prediction, decision-making, visual intelligence, etc. (Zhang). \\n\\tBCIs without artificial intelligence can perform, however a high level of data can’t be transferred efficiently without the help of a smart system. Advances have shown that AI performs better in translating neural signals and can help process signals sent from the brain on their way to the prostheses. AI in BCIs have algorithms that set internal parameters (like stimulation frequencies, electrical properties inside of the tissues in a neuron, pulse intervals, the amount of energy a device is using, etc. After collecting all this data, the AI model can sort through the data and only extract the useful information, classify it, and make sense out of the data by drawing patterns. This can aid a person to do various tasks such as hear, move, speak, and see (Zhang).\\n\\tRecently, more research has been committed to rehabilitation of the lower limbs using BCIs. The main idea is that it is possible to restore corticospinal and corticomuscular connections by correlating cortical activity of movement with intention. Implanted electrodes lets the user take control of movement on a higher and more complex degree (rather than just left, right, up, down). Using stochastic machine learning, an inbuilt classifier uses pregiven data to train the model so it can identify different categories of movements. The machine learning relies on deep learning which can classify the input features (which are varying brain frequency numbers) based on what action the user is doing in real time. Once it classifies which brain frequencies correlate to which actions, it can predict what the individual wants to do based on what brain frequency the individual is emitting (Zhang). \\n\\tThis section plans to explore the intersection between cognitive science and AI and how it can further augment research on communication and cognition. As described in previous sections, AI systems are built to perform in an intelligence, efficient, & time-saving manner that surpasses human performance. More research has been invested in creating an artificial intelligence model for hand gesture recognition, to benefit people not just with disabilities, but everyone. This research has made revolutionary technologies possible such as virtual reality, Smart TV, robot control, sign language recognition, etc. Without AI, the application situations are pretty limited as they don’t classify problems or movements efficiently',\n",
              "  '@search.score': 0.49963242},\n",
              " {'filepath': 'summary-31378681.txt',\n",
              "  'content': '1: Qiu Z, Li H, Zhang Z, Zhu Z, He S, Wang X, Wang P, Qin J, Zhuang L, Wang W,\\r\\nXie F, Gu Y, Zou K, Li C, Li C, Wang C, Cen J, Chen X, Shu Y, Zhang Z, Sun L,\\r\\nMin L, Fu Y, Huang X, Lv H, Zhou H, Ji Y, Zhang Z, Meng Z, Shi X, Zhang H, Li Y,\\r\\nHui L. A Pharmacogenomic Landscape in Human Liver Cancers. Cancer Cell. 2019 Aug\\r\\n12;36(2):179-193.e11. doi: 10.1016/j.ccell.2019.07.001. Epub 2019 Aug 1. PMID:\\r\\n31378681; PMCID: PMC7505724.',\n",
              "  '@search.score': 0.4767133},\n",
              " {'filepath': 'pubmed-31666728.txt',\n",
              "  'content': 'Gastroenterology & hepatology\\r\\nJID - 101500079\\r\\nRN - 0 (Biomarkers)\\r\\nSB - IM\\r\\nMH - Bile Duct Neoplasms/*metabolism\\r\\nMH - Biomarkers/metabolism\\r\\nMH - Chronic Disease\\r\\nMH - Disease Progression\\r\\nMH - Humans\\r\\nMH - Liver Diseases/complications/metabolism\\r\\nMH - Liver Neoplasms/etiology/*metabolism/pathology/physiopathology\\r\\nMH - Metabolic Networks and Pathways\\r\\nMH - Metabolomics\\r\\nMH - Prognosis\\r\\nMH - Systems Biology\\r\\nEDAT- 2019/11/02 06:00\\r\\nMHDA- 2020/02/11 06:00\\r\\nCRDT- 2019/11/01 06:00\\r\\nPHST- 2019/09/19 00:00 [accepted]\\r\\nPHST- 2019/11/02 06:00 [pubmed]\\r\\nPHST- 2020/02/11 06:00 [medline]\\r\\nPHST- 2019/11/01 06:00 [entrez]\\r\\nAID - 10.1038/s41575-019-0217-8 [pii]\\r\\nAID - 10.1038/s41575-019-0217-8 [doi]\\r\\nPST - ppublish\\r\\nSO - Nat Rev Gastroenterol Hepatol. 2019 Dec;16(12):748-766. doi: \\n 10.1038/s41575-019-0217-8. Epub 2019 Oct 30.',\n",
              "  '@search.score': 0.40088063}]"
            ]
          },
          "metadata": {},
          "execution_count": 5
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "# Semantic search result on top result for this question\n",
        "output = userquery(\"How does liver cancer originate\")[0][\"content\"]\n",
        "print(output)"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "kHUHtbtDiRwQ",
        "outputId": "6b112383-5e70-4533-de76-80623b4b19cb"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "PMID- 33080466\r\n",
            "OWN - NLM\r\n",
            "STAT- MEDLINE\r\n",
            "DCOM- 20210219\r\n",
            "LR - 20210219\r\n",
            "IS - 1950-6007 (Electronic)\r\n",
            "IS - 0753-3322 (Linking)\r\n",
            "VI - 132\r\n",
            "DP - 2020 Dec\r\n",
            "TI - WNT/β-catenin signaling in the development of liver cancers.\r\n",
            "PG - 110851\r\n",
            "LID - S0753-3322(20)31043-X [pii]\r\n",
            "LID - 10.1016/j.biopha.2020.110851 [doi]\r\n",
            "AB - The WNT/β-catenin signaling pathway is a highly conserved and tightly controlled \n",
            " molecular mechanism that regulates embryonic development, cellular proliferation \n",
            " and differentiation. Of note, accumulating evidence has shown that the aberrant \n",
            " of WNT/β-catenin signaling promotes the development and/or progression of liver \n",
            " cancer, including hepatocellular carcinoma (HCC) and cholangiocarcinoma (CCA), \n",
            " the two most prevalent primary liver tumours in adults. There are two different \n",
            " WNT signaling pathways have been identified, which were termed non-canonical and \n",
            " canonical pathways, the latter involving the activation of β-catenin. β-catenin, \n",
            " acting as an intracellular signal transducer in the WNT signaling pathway, is \n",
            " encoded by CTNNB1 and plays a critical role in tumorigenesis. In the past \n",
            " research, most liver tumors have mutations in genes encoding key components of \n",
            " the WNT/β-catenin signaling pathway. In addition, several of other signaling \n",
            " pathways also can crosswalk with β-catenin. In this review, we discuss the most \n",
            " relevant molecular mechanisms of action and regulation of WNT/β-catenin signaling \n",
            " in the development and pathophysiology of liver cancers, as well as in the \n",
            " development of therapeutics.\r\n",
            "CI - Copyright © 2020 The Author(s). Published by Elsevier Masson SAS.. All rights \n",
            " reserved.\r\n",
            "FAU - He, Shuai\r\n",
            "AU - He S\r\n",
            "AD - Department of General Surgery, the Fourth Affiliated Hospital, China Medical \n",
            " University, Shenyang 110032, China.\r\n",
            "FAU - Tang, Shilei\r\n",
            "AU - Tang S\r\n",
            "AD - Department of General Surgery, the Fourth Affiliated Hospital, China Medical \n",
            " University, Shenyang 110032, China. Electronic address: sltang@cmu.edu.cn.\r\n",
            "LA - eng\r\n",
            "PT - Journal Article\r\n",
            "PT - Review\r\n",
            "DEP - 20201017\r\n",
            "PL - France\r\n",
            "TA - Biomed Pharmacother\r\n",
            "JT - Biomedicine & pharmacotherapy = Biomedecine & pharmacotherapie\r\n",
            "JID - 8213295\r\n",
            "RN - 0 (CTNNB1 protein, human)\r\n",
            "RN - 0 (beta Catenin)\r\n",
            "SB - IM\r\n",
            "MH - Adult\r\n",
            "MH - Animals\r\n",
            "MH - Bile Duct Neoplasms/genetics/pathology\r\n",
            "MH - Carcinogenesis/genetics\r\n",
            "MH - Carcinoma, Hepatocellular/genetics/pathology\r\n",
            "MH - Cholangiocarcinoma/genetics\r\n",
            "MH - Humans\r\n",
            "MH - Liver Neoplasms/genetics/*pathology\r\n",
            "MH - Mutation\r\n",
            "MH - Wnt Signaling Pathway/*genetics\r\n",
            "MH - beta Catenin/*genetics\r\n",
            "OTO - NOTNLM\r\n",
            "OT - Liver cancer\r\n",
            "OT - Therapy\r\n",
            "OT - WNT/β-catenin\r\n",
            "EDAT- 2020/10/21 06:00\r\n",
            "MHDA- 2021/02/20 06:00\r\n",
            "CRDT- 2020/10/20 20:11\r\n",
            "PHST- 2020/08/26 00:00 [received]\r\n",
            "PHST- 2020/09/27 00:00 [revised]\r\n",
            "PHST- 2020/10/04 00:00 [accepted]\r\n",
            "PHST- 2020/10/21 06:00 [pubmed]\r\n",
            "PHST- 2021/02/20 06:00 [medline]\r\n",
            "PHST- 2020/10/20 20:11 [entrez]\r\n",
            "AID - S0753-3322(20)31043-X [pii]\r\n",
            "AID - 10.1016/j.biopha.2020.110851 [doi]\r\n",
            "PST - ppublish\r\n",
            "SO - Biomed Pharmacother\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!pip install openai"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "qEMrQaKMh7wb",
        "outputId": "19b82cea-d90b-4fbb-aa75-142840d2fb97"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Collecting openai\n",
            "  Downloading openai-0.27.8-py3-none-any.whl (73 kB)\n",
            "\u001b[?25l     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m0.0/73.6 kB\u001b[0m \u001b[31m?\u001b[0m eta \u001b[36m-:--:--\u001b[0m\r\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m73.6/73.6 kB\u001b[0m \u001b[31m2.9 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hRequirement already satisfied: requests>=2.20 in /usr/local/lib/python3.10/dist-packages (from openai) (2.31.0)\n",
            "Requirement already satisfied: tqdm in /usr/local/lib/python3.10/dist-packages (from openai) (4.65.0)\n",
            "Requirement already satisfied: aiohttp in /usr/local/lib/python3.10/dist-packages (from openai) (3.8.5)\n",
            "Requirement already satisfied: charset-normalizer<4,>=2 in /usr/local/lib/python3.10/dist-packages (from requests>=2.20->openai) (3.2.0)\n",
            "Requirement already satisfied: idna<4,>=2.5 in /usr/local/lib/python3.10/dist-packages (from requests>=2.20->openai) (3.4)\n",
            "Requirement already satisfied: urllib3<3,>=1.21.1 in /usr/local/lib/python3.10/dist-packages (from requests>=2.20->openai) (1.26.16)\n",
            "Requirement already satisfied: certifi>=2017.4.17 in /usr/local/lib/python3.10/dist-packages (from requests>=2.20->openai) (2023.7.22)\n",
            "Requirement already satisfied: attrs>=17.3.0 in /usr/local/lib/python3.10/dist-packages (from aiohttp->openai) (23.1.0)\n",
            "Requirement already satisfied: multidict<7.0,>=4.5 in /usr/local/lib/python3.10/dist-packages (from aiohttp->openai) (6.0.4)\n",
            "Requirement already satisfied: async-timeout<5.0,>=4.0.0a3 in /usr/local/lib/python3.10/dist-packages (from aiohttp->openai) (4.0.2)\n",
            "Requirement already satisfied: yarl<2.0,>=1.0 in /usr/local/lib/python3.10/dist-packages (from aiohttp->openai) (1.9.2)\n",
            "Requirement already satisfied: frozenlist>=1.1.1 in /usr/local/lib/python3.10/dist-packages (from aiohttp->openai) (1.4.0)\n",
            "Requirement already satisfied: aiosignal>=1.1.2 in /usr/local/lib/python3.10/dist-packages (from aiohttp->openai) (1.3.1)\n",
            "Installing collected packages: openai\n",
            "Successfully installed openai-0.27.8\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "import os\n",
        "import openai\n",
        "\n",
        "openai.api_key = \"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\""
      ],
      "metadata": {
        "id": "slydVdBolE1k"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "def get_completion_from_messages(messages,\n",
        "                                 model=\"gpt-3.5-turbo\",\n",
        "                                 temperature=0,\n",
        "                                 max_tokens=500):\n",
        "    response = openai.ChatCompletion.create(\n",
        "        model=model,\n",
        "        messages=messages,\n",
        "        temperature=temperature,\n",
        "        max_tokens=max_tokens,\n",
        "    )\n",
        "    return response.choices[0].message[\"content\"]"
      ],
      "metadata": {
        "id": "YaXSiHAThaog"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "\n",
        "delimiter = \"####\"\n",
        "system_message = f\"\"\"\n",
        "You will be provided with customer service queries. \\\n",
        "The customer service query will be delimited with \\\n",
        "{delimiter} characters.\n",
        "Classify each query into a primary category \\\n",
        "and a secondary category.\n",
        "Provide your output in json format with the \\\n",
        "keys: primary and secondary.\n",
        "\n",
        "Primary categories: Billing, Technical Support, \\\n",
        "Account Management, or General Inquiry.\n",
        "\n",
        "Billing secondary categories:\n",
        "Unsubscribe or upgrade\n",
        "Add a payment method\n",
        "Explanation for charge\n",
        "Dispute a charge\n",
        "\n",
        "Technical Support secondary categories:\n",
        "General troubleshooting\n",
        "Device compatibility\n",
        "Software updates\n",
        "\n",
        "Account Management secondary categories:\n",
        "Password reset\n",
        "Update personal information\n",
        "Close account\n",
        "Account security\n",
        "\n",
        "General Inquiry secondary categories:\n",
        "Product information\n",
        "Pricing\n",
        "Feedback\n",
        "Speak to a human\n",
        "\n",
        "\"\"\"\n",
        "user_message = f\"\"\"\\\n",
        "I want you to delete my profile and all of my user data\"\"\"\n",
        "messages =  [\n",
        "{'role':'system',\n",
        " 'content': system_message},\n",
        "{'role':'user',\n",
        " 'content': f\"{delimiter}{user_message}{delimiter}\"},\n",
        "]\n",
        "response = get_completion_from_messages(messages)\n",
        "print(response)"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "XDPQFFT5k7Co",
        "outputId": "09cdf4d9-bc2c-404f-e26c-19c284e5eebf"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "{\n",
            "  \"primary\": \"Account Management\",\n",
            "  \"secondary\": \"Close account\"\n",
            "}\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "def get_gpt_response_aslist(question):\n",
        "  response = openai.ChatCompletion.create(\n",
        "  engine=\"onlybasemodel\",\n",
        "  messages = [{\"role\":\"system\",\"content\":\"You are an AI assistant that helps people find information.\"},{\"role\":\"user\",\"content\":question}],\n",
        "  temperature=0,\n",
        "  max_tokens=800,\n",
        "  top_p=1,\n",
        "  frequency_penalty=0,\n",
        "  presence_penalty=0,\n",
        "  stop=None)\n",
        "  print([response['choices'][0][\"message\"][\"content\"]])\n",
        "  print(response)"
      ],
      "metadata": {
        "id": "PxuDTKfOhCeG"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "# Content of top 4 documents\n",
        "\n",
        "system_message = 'PMID- 33080466\\r\\nOWN - NLM\\r\\nSTAT- MEDLINE\\r\\nDCOM- 20210219\\r\\nLR - 20210219\\r\\nIS - 1950-6007 (Electronic)\\r\\nIS - 0753-3322 (Linking)\\r\\nVI - 132\\r\\nDP - 2020 Dec\\r\\nTI - WNT/β-catenin signaling in the development of liver cancers.\\r\\nPG - 110851\\r\\nLID - S0753-3322(20)31043-X [pii]\\r\\nLID - 10.1016/j.biopha.2020.110851 [doi]\\r\\nAB - The WNT/β-catenin signaling pathway is a highly conserved and tightly controlled \\n molecular mechanism that regulates embryonic development, cellular proliferation \\n and differentiation. Of note, accumulating evidence has shown that the aberrant \\n of WNT/β-catenin signaling promotes the development and/or progression of liver \\n cancer, including hepatocellular carcinoma (HCC) and cholangiocarcinoma (CCA), \\n the two most prevalent primary liver tumours in adults. There are two different \\n WNT signaling pathways have been identified, which were termed non-canonical and \\n canonical pathways, the latter involving the activation of β-catenin. β-catenin, \\n acting as an intracellular signal transducer in the WNT signaling pathway, is \\n encoded by CTNNB1 and plays a critical role in tumorigenesis. In the past \\n research, most liver tumors have mutations in genes encoding key components of \\n the WNT/β-catenin signaling pathway. In addition, several of other signaling \\n pathways also can crosswalk with β-catenin. In this review, we discuss the most \\n relevant molecular mechanisms of action and regulation of WNT/β-catenin signaling \\n in the development and pathophysiology of liver cancers, as well as in the \\n development of therapeutics.\\r\\nCI - Copyright © 2020 The Author(s). Published by Elsevier Masson SAS.. All rights \\n reserved.\\r\\nFAU - He, Shuai\\r\\nAU - He S\\r\\nAD - Department of General Surgery, the Fourth Affiliated Hospital, China Medical \\n University, Shenyang 110032, China.\\r\\nFAU - Tang, Shilei\\r\\nAU - Tang S\\r\\nAD - Department of General Surgery, the Fourth Affiliated Hospital, China Medical \\n University, Shenyang 110032, China. Electronic address: sltang@cmu.edu.cn.\\r\\nLA - eng\\r\\nPT - Journal Article\\r\\nPT - Review\\r\\nDEP - 20201017\\r\\nPL - France\\r\\nTA - Biomed Pharmacother\\r\\nJT - Biomedicine & pharmacotherapy = Biomedecine & pharmacotherapie\\r\\nJID - 8213295\\r\\nRN - 0 (CTNNB1 protein, human)\\r\\nRN - 0 (beta Catenin)\\r\\nSB - IM\\r\\nMH - Adult\\r\\nMH - Animals\\r\\nMH - Bile Duct Neoplasms/genetics/pathology\\r\\nMH - Carcinogenesis/genetics\\r\\nMH - Carcinoma, Hepatocellular/genetics/pathology\\r\\nMH - Cholangiocarcinoma/genetics\\r\\nMH - Humans\\r\\nMH - Liver Neoplasms/genetics/*pathology\\r\\nMH - Mutation\\r\\nMH - Wnt Signaling Pathway/*genetics\\r\\nMH - beta Catenin/*genetics\\r\\nOTO - NOTNLM\\r\\nOT - Liver cancer\\r\\nOT - Therapy\\r\\nOT - WNT/β-catenin\\r\\nEDAT- 2020/10/21 06:00\\r\\nMHDA- 2021/02/20 06:00\\r\\nCRDT- 2020/10/20 20:11\\r\\nPHST- 2020/08/26 00:00 [received]\\r\\nPHST- 2020/09/27 00:00 [revised]\\r\\nPHST- 2020/10/04 00:00 [accepted]\\r\\nPHST- 2020/10/21 06:00 [pubmed]\\r\\nPHST- 2021/02/20 06:00 [medline]\\r\\nPHST- 2020/10/20 20:11 [entrez]\\r\\nAID - S0753-3322(20)31043-X [pii]\\r\\nAID - 10.1016/j.biopha.2020.110851 [doi]\\r\\nPST - ppublish\\r\\nSO - Biomed Pharmacother'\n",
        "system_message2 = \"PMID- 35263795\\r\\nOWN - NLM\\r\\nSTAT- MEDLINE\\r\\nDCOM- 20220627\\r\\nLR - 20220630\\r\\nIS - 1098-8971 (Electronic)\\r\\nIS - 0272-8087 (Linking)\\r\\nVI - 42\\r\\nIP - 2\\r\\nDP - 2022 May\\r\\nTI - Liver Immunology, Immunotherapy, and Liver Cancers: Time for a Rethink?\\r\\nPG - 212-224\\r\\nLID - 10.1055/s-0042-1744143 [doi]\\r\\nAB - The complex immune system of the liver has a major role in tumor surveillance, \\n but also partly explains why current immune therapies are poorly effective \\n against liver cancers. Known primarily for its tolerogenic capacity, the hepatic \\n immune repertoire also comprises diverse populations of armored immune cells with \\n tumor surveillant roles. In healthy people, these work together to successfully \\n identify malignant cells and prevent their proliferation, thus halting tumor \\n formation. When frontline hepatic immune surveillance systems fail, compromised \\n hepatic immunity, driven by obesity, infection, or other pathological factors, \\n allows primary or secondary liver cancers to develop. Tumor growth promotes the \\n normal tolerogenic immunological milieu of the liver, perhaps explaining why \\n current immunotherapies fail to work. This review explores the complex local \\n liver immune system with the hope of identifying potential therapeutic targets \\n needed to best overcome immunological barriers in the liver to create an \\n environment no longer hostile to immunotherapy for the treatment of liver cancer.\\r\\nCI - Thieme. All rights reserved.\\r\\nFAU - Carroll, Hailey K\\r\\nAU - Carroll HK\\r\\nAD - Department of Medical Oncology, The Mater Hospital, Dublin, Ireland.\\r\\nFAU - Duffy, Austin G\\r\\nAU - Duffy AG\\r\\nAD - Department of Medical Oncology, The Mater Hospital, Dublin, Ireland.\\r\\nFAU - O'Farrelly, Cliona\\r\\nAU - O'Farrelly C\\r\\nAD - School of Biochemistry and Immunology, Trinity Biomedical Sciences Institute, \\n Trinity College, Dublin, Ireland.\\r\\nAD - School of Medicine, Trinity Biomedical Sciences Institute, Trinity College, \\n Dublin, Ireland.\\r\\nLA - eng\\r\\nPT - Journal Article\\r\\nPT - Review\\r\\nDEP - 20220309\\r\\nPL - United States\\r\\nTA - Semin Liver Dis\\r\\nJT - Seminars in liver disease\\r\\nJID - 8110297\\r\\nSB - IM\\r\\nMH - Humans\\r\\nMH - *Immunotherapy\\r\\nMH - *Liver Neoplasms/therapy\\r\\nMH - Tumor Microenvironment\\r\\nCOIS- None declared.\\r\\nEDAT- 2022/03/10 06:00\\r\\nMHDA- 2022/06/28 06:00\\r\\nCRDT- 2022/03/09 20:08\\r\\nPHST- 2022/03/10 06:00 [pubmed]\\r\\nPHST- 2022/06/28 06:00 [medline]\\r\\nPHST- 2022/03/09 20:08 [entrez]\\r\\nAID - 10.1055/s-0042-1744143 [doi]\\r\\nPST - ppublish\\r\\nSO - Semin Liver Dis. 2022 May;42(2):212-224. doi: 10.1055/s-0042-1744143. Epub 2022 \\n Mar 9.\"\n",
        "system_message3 = \"FAU - Zhang, Zhigang\\r\\nAU - Zhang Z\\r\\nAD - Department of Epidemiology and Biostatistics, Memorial Sloan-Kettering Cancer \\n Center, New York, NY 10065, USA.\\r\\nFAU - Meng, Zhiqiang\\r\\nAU - Meng Z\\r\\nAD - Department of Minimally Invasive Therapy, Collaborative Innovation Center for \\n Cancer Medicine, Fudan University Shanghai Cancer Center, Department of Oncology, \\n Shanghai Medical College, Fudan University, Shanghai 200032, China.\\r\\nFAU - Shi, Xiaolei\\r\\nAU - Shi X\\r\\nAD - Department of Hepatobiliary Surgery, the Affiliated Drum Tower Hospital, Medical \\n School of Nanjing University, Nanjing, Jiangsu 211166, China. Electronic address: \\n njsxl2000@163.com.\\r\\nFAU - Zhang, Haibin\\r\\nAU - Zhang H\\r\\nAD - Fifth Department of Hepatic Surgery, Eastern Hepatobiliary Surgery Hospital, \\n Second Military Medical University, Shanghai 200438, China. Electronic address: \\n drzhanghb@163.com.\\r\\nFAU - Li, Yixue\\r\\nAU - Li Y\\r\\nAD - CAS Key Laboratory of Computational Biology, CAS-MPG Partner Institute for \\n Computational Biology, Shanghai Institute of Nutrition and Health, Shanghai \\n Institutes for Biological Sciences, University of Chinese Academy of Sciences, \\n Chinese Academy of Sciences, Shanghai 200031, China. Electronic address: \\n yxli@sibs.ac.cn.\\r\\nFAU - Hui, Lijian\\r\\nAU - Hui L\\r\\nAD - State Key Laboratory of Cell Biology, CAS Center for Excellence in Molecular Cell \\n Science, Shanghai Institute of Biochemistry and Cell Biology, University of \\n Chinese Academy of Sciences, Chinese Academy of Sciences, Shanghai 200031, China; \\n School of Life Science and Technology, Shanghai Tech University, Shanghai 201210, \\n China; Bio-Research Innovation Center Suzhou, Shanghai Institute of Biochemistry \\n and Cell Biology, Chinese Academy of Sciences, Suzhou, Jiangsu 215121, China. \\n Electronic address: ljhui@sibcb.ac.cn.\\r\\nLA - eng\\r\\nGR - P30 CA008748/CA/NCI NIH HHS/United States\\r\\nPT - Journal Article\\r\\nPT - Research Support, Non-U.S. Gov't\\r\\nDEP - 20190801\\r\\nPL - United States\\r\\nTA - Cancer Cell\\r\\nJT - Cancer cell\\r\\nJID - 101130617\\r\\nRN - 0 (Antineoplastic Agents)\\r\\nRN - 0 (Biomarkers, Tumor)\\r\\nRN - 0 (Protein Kinase Inhibitors)\\r\\nRN - 9ZOQ3TZI87 (Sorafenib)\\r\\nSB - IM\\r\\nMH - Animals\\r\\nMH - Antineoplastic Agents/*pharmacology\\r\\nMH - Asian People/genetics\\r\\nMH - Biomarkers, Tumor/*genetics\\r\\nMH - Carcinoma, Hepatocellular/*drug therapy/ethnology/*genetics/pathology\\r\\nMH - Cell Line, Tumor\\r\\nMH - Clinical Decision-Making\\r\\nMH - Databases, Genetic\\r\\nMH - Drug Resistance, Neoplasm/genetics\\r\\nMH - Female\\r\\nMH - Genetic Heterogeneity\\r\\nMH - Genetic Predisposition to Disease\\r\\nMH - High-Throughput Nucleotide Sequencing\\r\\nMH - Humans\\r\\nMH - Liver Neoplasms/*drug therapy/ethnology/*genetics/pathology\\r\\nMH - Male\\r\\nMH - Mice, Inbred BALB C\\r\\nMH - Mice, Inbred NOD\\r\\nMH - Mice, Nude\\r\\nMH - Mice, SCID\\r\\nMH - Patient Selection\\r\\nMH - Pharmacogenomic Testing\\r\\nMH - *Pharmacogenomic Variants\\r\\nMH - Phenotype\\r\\nMH - Precision Medicine\\r\\nMH - Protein Kinase Inhibitors/*pharmacology\\r\\nMH - Sorafenib/*pharmacology\\r\\nMH - Xenograft Model Antitumor Assays\\r\\nPMC - PMC7505724\\r\\nMID - NIHMS1622926\\r\\nOTO - NOTNLM\\r\\nOT - liver cancer\\r\\nOT - patient-derived cancer models\\r\\nOT - pharmacogenomics\\r\\nOT - sorafenib\\r\\nCOIS- DECLARATION OF INTERESTS He Zhou, Wei Wang, Fubo Xie and Ying Gu are employees of \\n Shanghai ChemPartner Co., Ltd.\"\n",
        "system_message4 = 'PMID- 31378681\\r\\nOWN - NLM\\r\\nSTAT- MEDLINE\\r\\nDCOM- 20200420\\r\\nLR - 20221207\\r\\nIS - 1878-3686 (Electronic)\\r\\nIS - 1535-6108 (Print)\\r\\nIS - 1535-6108 (Linking)\\r\\nVI - 36\\r\\nIP - 2\\r\\nDP - 2019 Aug 12\\r\\nTI - A Pharmacogenomic Landscape in Human Liver Cancers.\\r\\nPG - 179-193.e11\\r\\nLID - S1535-6108(19)30328-9 [pii]\\r\\nLID - 10.1016/j.ccell.2019.07.001 [doi]\\r\\nAB - Liver cancers are highly heterogeneous with poor prognosis and drug response. A \\n better understanding between genetic alterations and drug responses would \\n facilitate precision treatment for liver cancers. To characterize the landscape \\n of pharmacogenomic interactions in liver cancers, we developed a protocol to \\n establish human liver cancer cell models at a success rate of around 50% and \\n generated the Liver Cancer Model Repository (LIMORE) with 81 cell models. LIMORE \\n represented genomic and transcriptomic heterogeneity of primary cancers. \\n Interrogation of the pharmacogenomic landscape of LIMORE discovered unexplored \\n gene-drug associations, including synthetic lethalities to prevalent alterations \\n in liver cancers. Moreover, predictive biomarker candidates were suggested for \\n the selection of sorafenib-responding patients. LIMORE provides a rich resource \\n facilitating drug discovery in liver cancers.\\r\\nCI - Copyright © 2019 Elsevier Inc. All rights reserved.\\r\\nFAU - Qiu, Zhixin\\r\\nAU - Qiu Z\\r\\nAD - State Key Laboratory of Cell Biology, CAS Center for Excellence in Molecular Cell \\n Science, Shanghai Institute of Biochemistry and Cell Biology, University of \\n Chinese Academy of Sciences, Chinese Academy of Sciences, Shanghai 200031, China.\\r\\nFAU - Li, Hong\\r\\nAU - Li H\\r\\nAD - CAS Key Laboratory of Computational Biology, CAS-MPG Partner Institute for \\n Computational Biology, Shanghai Institute of Nutrition and Health, Shanghai \\n Institutes for Biological Sciences, University of Chinese Academy of Sciences, \\n Chinese Academy of Sciences, Shanghai 200031, China.\\r\\nFAU - Zhang, Zhengtao\\r\\nAU - Zhang Z\\r\\nAD - State Key Laboratory of Cell Biology, CAS Center for Excellence in Molecular Cell \\n Science, Shanghai Institute of Biochemistry and Cell Biology, University of \\n Chinese Academy of Sciences, Chinese Academy of Sciences, Shanghai 200031, China.\\r\\nFAU - Zhu, Zhenfeng\\r\\nAU - Zhu Z\\r\\nAD - Department of Minimally Invasive Therapy, Collaborative Innovation Center for \\n Cancer Medicine, Fudan University Shanghai Cancer Center, Department of Oncology, \\n Shanghai Medical College, Fudan University, Shanghai 200032, China.\\r\\nFAU - He, Sheng\\r\\nAU - He S\\r\\nAD - CAS Key Laboratory of Computational Biology, CAS-MPG Partner Institute for \\n Computational Biology, Shanghai Institute of Nutrition and Health, Shanghai \\n Institutes for Biological Sciences, University of Chinese Academy of Sciences, \\n Chinese Academy of Sciences, Shanghai 200031, China; School of Life Science and \\n Technology, Shanghai Tech University, Shanghai 201210, China.\\r\\nFAU - Wang, Xujun\\r\\nAU - Wang X\\r\\nAD - SJTU-Yale Joint Center for Biostatistics, Department of Bioinformatics and \\n Biostatistics, Shanghai Jiaotong University, Shanghai 200240, China.\\r\\nFAU - Wang, Pengcheng\\r\\nAU - Wang P\\r\\nAD - Liver Cancer Institute, Zhongshan Hospital, Fudan University, Key Laboratory of \\n Carcinogenesis and Cancer Invasion, Fudan University, Shanghai 200032, China.\\r\\nFAU - Qin, Jianjie\\r\\nAU - Qin J\\r\\nAD - Liver Transplantation Center, Key Laboratory of Living Donor Liver \\n Transplantation of Ministry of Public Health, The First Affiliated Hospital of \\n Nanjing Medical University, Nanjing, Jiangsu 210029, China.\\r\\nFAU - Zhuang, Liping\\r\\nAU - Zhuang L\\r\\nAD - Department of Minimally Invasive Therapy, Collaborative Innovation Center for \\n Cancer Medicine, Fudan University Shanghai Cancer Center, Department of Oncology, \\n Shanghai Medical College, Fudan University, Shanghai 200032, China.\\r\\nFAU - Wang, Wei\\r\\nAU - Wang W\\r\\nAD - Shanghai ChemPartner Co., Ltd., Shanghai 201203, China'\n",
        "x = system_message+system_message2+system_message3+system_message4\n"
      ],
      "metadata": {
        "id": "SJK6PER5iA_p"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "delimiter = \"####\"\n",
        "x\n",
        "user_message = f\"\"\"\\Please summarize the user content into 5 sentences.\n",
        "Also print the token count.\n",
        "If it exceeds token count, just say 'Input exceeded token count'\n",
        "\"\"\"\n",
        "\n",
        "messages =  [\n",
        "{'role':'system',\n",
        " 'content': user_message},\n",
        "{'role':'user',\n",
        " 'content': f\"{delimiter}{x}{delimiter}\"},\n",
        "]\n",
        "response = get_completion_from_messages(messages)\n",
        "print(response)"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "XZ8ubEGPC9Lu",
        "outputId": "2d4a47d5-9380-42e3-dd28-6b895d18f3d7"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "The first article discusses the role of WNT/β-catenin signaling in the development of liver cancers. The WNT/β-catenin pathway is involved in embryonic development and cellular proliferation and differentiation. Aberrant activation of this pathway has been shown to promote the development and progression of liver cancer. The review discusses the molecular mechanisms of WNT/β-catenin signaling in liver cancer development and therapeutics.\n",
            "\n",
            "The second article explores the complex immune system of the liver and its role in tumor surveillance. The liver has a tolerogenic capacity but also contains immune cells that can identify and prevent the proliferation of malignant cells. However, compromised hepatic immunity can allow liver cancers to develop. The review discusses the immune system of the liver and potential therapeutic targets to overcome immunological barriers in liver cancer treatment.\n",
            "\n",
            "The third article focuses on the pharmacogenomic landscape of human liver cancers. The researchers developed a protocol to establish human liver cancer cell models and created a repository of these models. They used these models to study the interactions between genetic alterations and drug responses in liver cancer. The study identified gene-drug associations and suggested predictive biomarker candidates for the selection of patients who will respond to sorafenib treatment.\n",
            "\n",
            "The token count is 446.\n"
          ]
        }
      ]
    }
  ]
}